<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: Interactively Filter SNP Datasets</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">
<h1> Interactively Filter SNP Datasets
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;SNPfiltR&rsquo; version 0.0.0.9000</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
</ul>

<h2>Help Pages</h2>


<table width="100%">
<tr><td style="width: 25%;"><a href="assess_missing_data_pca.html">assess_missing_data_pca</a></td>
<td>Vizualise how missing data thresholds affect sample clustering</td></tr>
<tr><td style="width: 25%;"><a href="assess_missing_data_tsne.html">assess_missing_data_tsne</a></td>
<td>Vizualise how missing data thresholds affect sample clustering</td></tr>
<tr><td style="width: 25%;"><a href="distance_thin.html">distance_thin</a></td>
<td>Filter a vcf file based on distance between SNPs on a given scaffold</td></tr>
<tr><td style="width: 25%;"><a href="filter_allele_balance.html">filter_allele_balance</a></td>
<td>Filter out heterozygous genotypes failing an allele balance check</td></tr>
<tr><td style="width: 25%;"><a href="filter_biallelic.html">filter_biallelic</a></td>
<td>Remove SNPs with more than two alleles</td></tr>
<tr><td style="width: 25%;"><a href="hard_filter.html">hard_filter</a></td>
<td>Hard filter a vcf file by depth and genotype quality (gq)</td></tr>
<tr><td style="width: 25%;"><a href="max_depth.html">max_depth</a></td>
<td>Vizualise and filter based on mean depth across all called SNPs</td></tr>
<tr><td style="width: 25%;"><a href="min_mac.html">min_mac</a></td>
<td>Vizualise, filter based on Minor Allele Count (MAC)</td></tr>
<tr><td style="width: 25%;"><a href="missing_by_sample.html">missing_by_sample</a></td>
<td>Vizualise missing data per sample, remove samples above a missing data cutoff</td></tr>
<tr><td style="width: 25%;"><a href="missing_by_snp.html">missing_by_snp</a></td>
<td>Vizualise missing data per SNP, remove SNPs above a missing data cutoff</td></tr>
<tr><td style="width: 25%;"><a href="popmap.html">popmap</a></td>
<td>Popmap for example scrub-jay vcfR file</td></tr>
<tr><td style="width: 25%;"><a href="vcfR.example.html">vcfR.example</a></td>
<td>Example scrub-jay vcfR file</td></tr>
</table>
</div></body></html>
