---
title: "accuracy-benchmarking"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{accuracy-benchmarking}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

This vignette is designed to validate the accuracy of the key functions from the SNPfiltR package against the established VCFtools program. For each function, I filtered 3 vcf files according to the same parameter thresholds using SNPfiltR and VCFtools, and then tested whether each genotype in the output vcf files matched. All functions matched exactly, except for the distance_thin() function. It seems that the SNPfiltR approach accurately thinned the dataset, while the '--thin 100' flag from VCFtools allowed SNPs to remain in the output dataset despite being within 100 base pairs of nearby SNPs. I am not sure why VCFtools is less accurate on these vcf files I used for testing, and encourage more testing from other users in case this is an idiosyncratic issue with the particular vcf files I used. Meanwhile, I suggest using the SNPfiltR function distance_thin() which is validated clearly here, over the VCFtools approach.

All of the input vcf files necessary to fully reproduce this accuracy comparison can be found [here](https://github.com/DevonDeRaad/SNPfiltR.benchmarking.data)

```{r}
library(SNPfiltR)
library(vcfR)
library(microbenchmark)
library(ggplot2)
library(reshape2)
```

# Validate the hard_filter() function

### Filter each vcf with SNPfiltR
```{r}
#perform filtering on the 10K SNP vcf file and save the result
r.10k<-hard_filter(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"),depth = 5,gq = 30)
r.20k<-hard_filter(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"),depth = 5,gq = 30)
r.50k<-hard_filter(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"),depth = 5,gq = 30)
```

### Filter each with VCFtools
```{bash}
cd /Users/devder/Desktop/benchmarking.vcfs
/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.10K.vcf.gz --minGQ 30 --minDP 5 --recode --recode-INFO-all --out 10K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.20K.vcf.gz --minGQ 30 --minDP 5 --recode --recode-INFO-all --out 20K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.50K.vcf.gz --minGQ 30 --minDP 5 --recode --recode-INFO-all --out 50K.filtered
```

### Read in VCFtools files
```{r}
#read in all the vcftools files
bash.10k<-read.vcfR("~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf")
bash.20k<-read.vcfR("~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf")
bash.50k<-read.vcfR("~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf")
```

### check that every cell in the genotpye matrix matches for each file
```{r}
#Does every cell in the genotype matrix for each camparable vcf match?
table(r.10k@gt == bash.10k@gt)
table(r.20k@gt == bash.20k@gt)
table(r.50k@gt == bash.50k@gt)
```

# Validate the missing_by_snp() function

### Filter each vcf with SNPfiltR
```{r}
#perform filtering on each vcf file and save the result
r.10k<-missing_by_snp(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"),cutoff=.5)
r.20k<-missing_by_snp(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"),cutoff=.5)
r.50k<-missing_by_snp(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"),cutoff=.5)
```

### Filter each with VCFtools
```{bash}
cd /Users/devder/Desktop/benchmarking.vcfs
/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.10K.vcf.gz --max-missing 0.5 --recode --recode-INFO-all --out 10K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.20K.vcf.gz --max-missing 0.5 --recode --recode-INFO-all --out 20K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.50K.vcf.gz --max-missing 0.5 --recode --recode-INFO-all --out 50K.filtered
```

### Read in VCFtools files
```{r}
#read in all the vcftools files
bash.10k<-read.vcfR("~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf")
bash.20k<-read.vcfR("~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf")
bash.50k<-read.vcfR("~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf")
```

### check that every cell in the genotpye matrix matches for each file
```{r}
#Does every cell in the genotype matrix for each camparable vcf match?
table(r.10k@gt == bash.10k@gt)
table(r.20k@gt == bash.20k@gt)
table(r.50k@gt == bash.50k@gt)
```
# Validate the min_mac() function

### Filter each vcf with SNPfiltR
```{r}
#perform filtering on each vcf file and save the result
r.10k<-min_mac(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"),min.mac = 2)
r.20k<-min_mac(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"),min.mac = 2)
r.50k<-min_mac(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"),min.mac = 2)
```

### Filter each with VCFtools
```{bash}
cd /Users/devder/Desktop/benchmarking.vcfs
/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.10K.vcf.gz --mac 2 --recode --recode-INFO-all --out 10K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.20K.vcf.gz --mac 2 --recode --recode-INFO-all --out 20K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.50K.vcf.gz --mac 2 --recode --recode-INFO-all --out 50K.filtered
```

### Read in VCFtools files
```{r}
#read in all the vcftools files
bash.10k<-read.vcfR("~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf")
bash.20k<-read.vcfR("~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf")
bash.50k<-read.vcfR("~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf")
```

### check that every cell in the genotpye matrix matches for each file
```{r}
#Does every cell in the genotype matrix for each camparable vcf match?
table(r.10k@gt == bash.10k@gt)
table(r.20k@gt == bash.20k@gt)
table(r.50k@gt == bash.50k@gt)
```

# Validate the distance_thin() function

### Filter each vcf with SNPfiltR
```{r}
#perform filtering on each vcf file and save the result
r.10k<-distance_thin(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"),min.distance=100)
r.20k<-distance_thin(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"),min.distance=100)
r.50k<-distance_thin(vcfR = read.vcfR("~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"),min.distance=100)
```

### Filter each with VCFtools
```{bash}
cd /Users/devder/Desktop/benchmarking.vcfs
/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.10K.vcf.gz --thin 100 --recode --recode-INFO-all --out 10K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.20K.vcf.gz --thin 100 --recode --recode-INFO-all --out 20K.filtered

/Users/devder/Downloads/vcftools/src/cpp/vcftools --gzvcf benchmark.50K.vcf.gz --thin 100 --recode --recode-INFO-all --out 50K.filtered
```

### Read in VCFtools files
```{r}
#read in all the vcftools files
bash.10k<-read.vcfR("~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf")
bash.20k<-read.vcfR("~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf")
bash.50k<-read.vcfR("~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf")
```

### check that every cell in the genotpye matrix matches for each file
```{r}
#Does every cell in the genotype matrix for each camparable vcf match?
dim(r.10k@gt)
dim(bash.10k@gt)

dim(r.20k@gt)
dim(bash.20k@gt)

dim(r.50k@gt)
dim(bash.50k@gt)
#right away we can see that the dimensions of these vcf files don't match, indicating that the results differ.

#Let's calculate distance between SNPs in each vcf by hand to make sure both methods work as promised.

### Start with SNPfiltR thinned vcf:
##
#

#generate dataframe containing information for chromosome and bp locality of each SNP
df<-as.data.frame(r.10k@fix[,1:2])
#generate list of all unique chromosomes in alphabetical order
chroms<-levels(as.factor(df$CHROM))
#intialize empty df to hold filtering
keep.df<-data.frame()
#loop over each chromosome
#for t in vector containing the name of each chromosome
for (t in chroms){

  #isolate the SNP positions on the given chromosome
  fix.sub<-as.numeric(df$POS[df$CHROM == t])

  #order the positions numerically
  fix.sub<-fix.sub[order(fix.sub)]

  #set the first position
  prev<-fix.sub[1]

  #initialize empty vector
  k<-c()
  k[1]<-NA
  
  #loop to decide whether to keep each following SNP
    if (length(fix.sub) < 2){
      #if chrom only has 1 SNP, do nothing
    } else{

      #else, use a for loop to determine which SNPs to keep that satisfy our distance criteria
      for (i in 2:length(fix.sub)){

        #store distance in base pairs of the current SNP from the previous SNP
        k[i]<- fix.sub[i] - prev

        #update the current SNP to be the previous SNP for the next iteration of this loop
        prev<-fix.sub[i]
        
      #close for loop
      }
      #k[1]<- fix.sub[2] - fix.sub[1]
    #close else statement
    }

  #make a dataframe with the precise info for each SNP for this chromosome
  chrom.df<-data.frame(CHROM=rep(t, times=length(fix.sub)), POS=fix.sub, dist=k)

  #now we rbind in the information for this chromosome to the overall df
  keep.df<-rbind(keep.df,chrom.df)

  #empty df for this chrom to prepare for the next one
  chrom.df<-NULL

} #close for loop, start over on next chromosome

#order the dataframe to match the order of the input vcf file
order.df<-keep.df[match(paste(df$CHROM,df$POS), paste(keep.df$CHROM,keep.df$POS)),]
order.df<-order.df[order(order.df$dist),]

#show the lower distribution of SNP distances to see if any fall below the 100bp threshold
hist(order.df$dist, xlim =c(0,1000), breaks = 50000, main = "thinning using SNPfiltR",
          xlab = "Distance from nearest SNP (base pairs)")
abline(v=100, col="red")

head(order.df) #show the tail of SNP distances

#store df
snpfilt.df<-order.df

### repeat for the VCFtools filtered vcf
##
#

#generate dataframe containing information for chromosome and bp locality of each SNP
df<-as.data.frame(bash.10k@fix[,1:2])
#generate list of all unique chromosomes in alphabetical order
chroms<-levels(as.factor(df$CHROM))
#intialize empty df to hold filtering
keep.df<-data.frame()
#loop over each chromosome
#for t in vector containing the name of each chromosome
for (t in chroms){

  #isolate the SNP positions on the given chromosome
  fix.sub<-as.numeric(df$POS[df$CHROM == t])

  #order the positions numerically
  fix.sub<-fix.sub[order(fix.sub)]

  #set the first position
  prev<-fix.sub[1]

  #initialize empty vector
  k<-c()
  k[1]<-NA
  
  #loop to decide whether to keep each following SNP
    if (length(fix.sub) < 2){
      #if chrom only has 1 SNP, do nothing
    } else{

      #else, use a for loop to determine which SNPs to keep that satisfy our distance criteria
      for (i in 2:length(fix.sub)){

        #store distance in base pairs of the current SNP from the previous SNP
        k[i]<- fix.sub[i] - prev

        #update the current SNP to be the previous SNP for the next iteration of this loop
        prev<-fix.sub[i]
        
      #close for loop
      }
      #k[1]<- fix.sub[2] - fix.sub[1]
    #close else statement
    }

  #make a dataframe with the precise info for each SNP for this chromosome
  chrom.df<-data.frame(CHROM=rep(t, times=length(fix.sub)), POS=fix.sub, dist=k)

  #now we rbind in the information for this chromosome to the overall df
  keep.df<-rbind(keep.df,chrom.df)

  #empty df for this chrom to prepare for the next one
  chrom.df<-NULL

} #close for loop, start over on next chromosome

#order the dataframe to match the order of the input vcf file
order.df<-keep.df[match(paste(df$CHROM,df$POS), paste(keep.df$CHROM,keep.df$POS)),]
order.df<-order.df[order(order.df$dist),]

hist(order.df$dist, xlim =c(0,1000), breaks = 50000, main = "thinning using VCFtools",
     xlab = "Distance from nearest SNP (base pairs)")
abline(v=100, col="red")
head(order.df) #show the tail of SNP distances

#putting the dataframe in chromosomal order, we can see right away that SNPs less than 100bp apart have been retained
order.df<-order.df[with(order.df, order(CHROM, POS)),]
order.df[1:10,]

#plot the two together on the same histogram
order.df$approach<-rep("VCFtools", times=nrow(order.df))
snpfilt.df$approach<-rep("SNPfiltR", times=nrow(snpfilt.df))
sum.df<-rbind(order.df,snpfilt.df)

#plot both
ggplot(sum.df, aes(x=dist, color=approach)) +
  geom_histogram(bins=100, fill="white")+
  theme_classic()+
  xlim(c(0, 1000))+
  xlab(label = "Distance from nearest SNP (base pairs)")+
  geom_vline(xintercept = 100, linetype="dashed")+
  theme(legend.position = c(0.8, 0.8))

hists<-ggplot(sum.df, aes(x=dist, color=approach)) +
  geom_histogram(bins=100, fill="white")+
  theme_classic()+
  xlim(c(0, 2500))+
  xlab(label = "Distance from nearest SNP (base pairs)")+
  geom_vline(xintercept = 100, linetype="dashed")+
  theme(legend.position = c(0.8, 0.8))

#save plot
ggsave(hists,
      filename = "~/Desktop/SNPfiltR.mol.ecol.resour.submission/resubmission/validate.thinning.pdf",
      height = 2, width = 8, units = "in")

#see how many retained SNPs overlap between the two approaches for each file
table(with(as.data.frame(r.10k@fix), paste0(CHROM,POS)) %in% with(as.data.frame(bash.10k@fix), paste0(CHROM,POS)))
table(with(as.data.frame(r.20k@fix), paste0(CHROM,POS)) %in% with(as.data.frame(bash.20k@fix), paste0(CHROM,POS)))
table(with(as.data.frame(r.50k@fix), paste0(CHROM,POS)) %in% with(as.data.frame(bash.50k@fix), paste0(CHROM,POS)))

#store number of SNPs for each approach as a vector
r.snps<-c()
r.snps[1]<-nrow(r.10k@gt)
r.snps[2]<-nrow(r.20k@gt)
r.snps[3]<-nrow(r.50k@gt)

#store number of bash SNPs
bash.snps<-c()
bash.snps[1]<-nrow(bash.10k@gt)
bash.snps[2]<-nrow(bash.20k@gt)
bash.snps[3]<-nrow(bash.50k@gt)

bar.frame<-data.frame(snps=c(r.snps,bash.snps),
                      approach=c("SNPfiltR::distance_thin()","SNPfiltR::distance_thin()","SNPfiltR::distance_thin()",
                               "VCFtools --thin","VCFtools --thin","VCFtools --thin"),
                      input.snps=c("10K","20K","50K",
                                   "10K","20K","50K")
                      )

## Grouped bar plot
ggplot(bar.frame, aes(fill=approach, y=snps, x=input.snps)) + 
  geom_bar(position="dodge", stat="identity")+
  xlab("SNPs in input vcf file")+
  ylab("Number of thinned SNPs")+
  theme_classic()

thin.bar<-ggplot(bar.frame, aes(fill=approach, y=snps, x=input.snps)) + 
  geom_bar(position="dodge", stat="identity")+
  xlab("SNPs in input vcf file")+
  ylab("Number of thinned SNPs")+
  theme_classic()+
  theme(legend.position = c(0.3, 0.8))

#save plot
ggsave(thin.bar,
      filename = "~/Desktop/SNPfiltR.mol.ecol.resour.submission/resubmission/thinning.comp.pdf",
      height = 3, width = 4.5, units = "in")

```

### Make final table showing genotpye concordance across comparisons
```{r}
# print table showing concordance
concord<-data.frame(concordance=c(1,1,1,1,1,1,1,1,1,
                                  sum(with(as.data.frame(r.10k@fix), paste0(CHROM,POS)) %in% with(as.data.frame(bash.10k@fix), paste0(CHROM,POS)))/nrow(as.data.frame(r.10k@fix)),
                                  sum(with(as.data.frame(r.20k@fix), paste0(CHROM,POS)) %in% with(as.data.frame(bash.20k@fix), paste0(CHROM,POS)))/nrow(as.data.frame(r.20k@fix)),
                                  sum(with(as.data.frame(r.50k@fix), paste0(CHROM,POS)) %in% with(as.data.frame(bash.50k@fix), paste0(CHROM,POS)))/nrow(as.data.frame(r.50k@fix))),
                      filtering=c("GQ","GQ","GQ","missing data","missing data","missing data",
                                  "MAC","MAC","MAC","distance thin","distance thin","distance thin"),
                      input.snps=c("10K","20K","50K",
                                   "10K","20K","50K",
                                   "10K","20K","50K",
                                   "10K","20K","50K")
                      )

#print table
heat <- reshape2::melt(concord)
ggplot(data = heat, aes(x=input.snps, y=filtering, fill=value)) + 
  geom_tile(color = "white", size=1)+
  geom_text(data=heat,aes(label=round(value, 2)))+
  theme_minimal()+
  scale_fill_gradient2(low = "white", high = "cornflowerblue", space = "Lab", name="genotype\nconcordance")+
  ylab("type of filtering")+
  xlab("Number of SNPs in input vcf file")

geno.concord.plot<-ggplot(data = heat, aes(x=input.snps, y=filtering, fill=value)) + 
  geom_tile(color = "white", size=1)+
  geom_text(data=heat,aes(label=round(value, 2)))+
  theme_minimal()+
  scale_fill_gradient2(low = "white", high = "cornflowerblue", space = "Lab", name="genotype\nconcordance")+
  ylab("type of filtering")+
  xlab("Number of SNPs in input vcf file")

#save plot
#ggsave(geno.concord.plot,
#      filename = "~/Desktop/SNPfiltR.mol.ecol.resour.submission/resubmission/geno.concord.pdf",
#      height = 3, width = 5, units = "in")
```

