<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accuracy benchmarking • SNPfiltR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accuracy benchmarking">
<meta property="og:description" content="SNPfiltR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/accuracy-benchmarking.html">Accuracy benchmarking</a>
    </li>
    <li>
      <a href="../articles/performance-benchmarking.html">Performance benchmarking</a>
    </li>
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-RADseq-vignette.html">scrub-jay-RADseq-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-UCE-vignette.html">scrub-jay-UCE-vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="accuracy-benchmarking_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accuracy benchmarking</h1>
                        <h4 class="author">Devon DeRaad</h4>
            
            <h4 class="date">2/28/2022</h4>
      
      
      <div class="hidden name"><code>accuracy-benchmarking.Rmd</code></div>

    </div>

    
    
<p>This vignette is designed to validate the accuracy of the key functions from the SNPfiltR package against the established VCFtools program. For each function, I filtered 3 vcf files according to the same parameter thresholds using SNPfiltR and VCFtools, and then tested whether each genotype in the output vcf files matched. All functions matched exactly, except for the distance_thin() function. It seems that the SNPfiltR approach accurately thinned the dataset, while the ‘–thin 100’ flag from VCFtools allowed SNPs to remain in the output dataset despite being within 100 base pairs of nearby SNPs. I am not sure why VCFtools is less accurate on these vcf files I used for testing, and encourage more testing from other users in case this is an idiosyncratic issue with the particular vcf files I used. Meanwhile, I suggest using the SNPfiltR function distance_thin() which is validated clearly here, over the VCFtools approach.</p>
<p>All of the input vcf files necessary to fully reproduce this accuracy comparison can be found <a href="github.com/DevonDeRaad/SNPfiltR.benchmarking.data">here</a></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SNPfiltR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR">vcfR</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
##    *****       ***   vcfR   ***       *****
##    This is vcfR 1.12.0 
##      browseVignettes('vcfR') # Documentation
##      citation('vcfR') # Citation
##    *****       *****      *****       *****</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joshuaulrich/microbenchmark/">microbenchmark</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></code></pre></div>
<div id="validate-the-hard_filter-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-hard_filter-function" class="anchor"></a>Validate the hard_filter() function</h1>
<div id="filter-each-vcf-with-snpfiltr" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on the 10K SNP vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<pre><code>## 28.84% of genotypes fall below a read depth of 5 and were converted to NA</code></pre>
<pre><code>## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<pre><code>## 29.06% of genotypes fall below a read depth of 5 and were converted to NA</code></pre>
<pre><code>## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<pre><code>## 29.34% of genotypes fall below a read depth of 5 and were converted to NA</code></pre>
<pre><code>## 1.33% of genotypes fall below a genotype quality of 30 and were converted to NA</code></pre>
</div>
<div id="filter-each-with-vcftools" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --minDP 5
##  --minGQ 30
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 10000 out of a possible 10000 Sites
## Run Time = 2.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --minDP 5
##  --minGQ 30
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 20000 out of a possible 20000 Sites
## Run Time = 2.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --minDP 5
##  --minGQ 30
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 50000 out of a possible 50000 Sites
## Run Time = 6.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 160694</code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 316967</code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 781671</code></pre>
</div>
</div>
<div id="validate-the-missing_by_snp-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-missing_by_snp-function" class="anchor"></a>Validate the missing_by_snp() function</h1>
<div id="filter-each-vcf-with-snpfiltr-1" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr-1" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on each vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,cutoff<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<pre><code>## cutoff is specified, filtered vcfR object will be returned</code></pre>
<pre><code>## 80.93% of SNPs fell below a completeness cutoff of 0.5 and were removed from the VCF</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,cutoff<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<pre><code>## cutoff is specified, filtered vcfR object will be returned</code></pre>
<pre><code>## 81.22% of SNPs fell below a completeness cutoff of 0.5 and were removed from the VCF</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-6-2.png" width="700"></p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,cutoff<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<pre><code>## cutoff is specified, filtered vcfR object will be returned</code></pre>
<pre><code>## 81.51% of SNPs fell below a completeness cutoff of 0.5 and were removed from the VCF</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-6-3.png" width="700"></p>
</div>
<div id="filter-each-with-vcftools-1" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools-1" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--max-missing</span> 0.5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--max-missing</span> 0.5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--max-missing</span> 0.5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --max-missing 0.5
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 1907 out of a possible 10000 Sites
## Run Time = 0.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --max-missing 0.5
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 3757 out of a possible 20000 Sites
## Run Time = 1.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --max-missing 0.5
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 9243 out of a possible 50000 Sites
## Run Time = 3.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files-1" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files-1" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 1907
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 1907
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 1907
##   row_num: 0
## 
Processed variant 1000
Processed variant: 1907
## All variants processed</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 3757
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 3757
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 3757
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant: 3757
## All variants processed</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 9243
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 9243
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 9243
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant: 9243
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-1" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-1" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 160911</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 316502</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 779516</code></pre>
</div>
</div>
<div id="validate-the-distance_thin-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-distance_thin-function" class="anchor"></a>Validate the distance_thin() function</h1>
<div id="filter-each-vcf-with-snpfiltr-2" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr-2" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on each vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/distance_thin.html">distance_thin</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,min.distance<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |=====                                                                 |   6%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |==============                                                        |  19%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |=======================                                               |  32%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |===========================                                           |  39%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |===========================================                           |  61%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |===============================================                       |  68%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  81%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |=================================================================     |  94%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code>## 9399 out of 10000 input SNPs were not located within 100 base-pairs of another SNP and were retained despite filtering</code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/distance_thin.html">distance_thin</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,min.distance<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |==========                                                            |  15%
  |                                                                            
  |============                                                          |  18%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |================                                                      |  24%
  |                                                                            
  |===================                                                   |  26%
  |                                                                            
  |=====================                                                 |  29%
  |                                                                            
  |=======================                                               |  32%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |===============================================                       |  68%
  |                                                                            
  |=================================================                     |  71%
  |                                                                            
  |===================================================                   |  74%
  |                                                                            
  |======================================================                |  76%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |==========================================================            |  82%
  |                                                                            
  |============================================================          |  85%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code>## 17745 out of 20000 input SNPs were not located within 100 base-pairs of another SNP and were retained despite filtering</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/distance_thin.html">distance_thin</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,min.distance<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code>## 38371 out of 50000 input SNPs were not located within 100 base-pairs of another SNP and were retained despite filtering</code></pre>
</div>
<div id="filter-each-with-vcftools-2" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools-2" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb65"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--thin</span> 100 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--thin</span> 100 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--thin</span> 100 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --thin 100
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 9464 out of a possible 10000 Sites
## Run Time = 1.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --thin 100
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 18947 out of a possible 20000 Sites
## Run Time = 3.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --thin 100
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 47218 out of a possible 50000 Sites
## Run Time = 5.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files-2" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files-2" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 9464
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 9464
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 9464
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant: 9464
## All variants processed</code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 18947
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 18947
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 18947
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant: 18947
## All variants processed</code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 47218
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 47218
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 47218
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant: 47218
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-2" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-2" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 9399  101</code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 9464  101</code></pre>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#right away we can see that the dimensions of these vcf files don't match, indicating that the results differ.</span>

<span class="co">#Let'scalculate distance between SNPs in each vcf by hand to see who was more accurate.</span>

<span class="co">### Start with SNPfiltR thinned vcf:</span>
<span class="co">##</span>
<span class="co">#</span>

<span class="co">#generate dataframe containing information for chromosome and bp locality of each SNP</span>
<span class="va">df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#generate list of all unique chromosomes in alphabetical order</span>
<span class="va">chroms</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span><span class="op">)</span><span class="op">)</span>
<span class="co">#intialize empty df to hold filtering</span>
<span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#loop over each chromosome</span>
<span class="co">#for t in vector containing the name of each chromosome</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">t</span> <span class="kw">in</span> <span class="va">chroms</span><span class="op">)</span><span class="op">{</span>

  <span class="co">#isolate the SNP positions on the given chromosome</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span> <span class="op">==</span> <span class="va">t</span><span class="op">]</span><span class="op">)</span>

  <span class="co">#order the positions numerically</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">]</span>

  <span class="co">#set the first position</span>
  <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

  <span class="co">#initialize empty vector</span>
  <span class="va">k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">k</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="cn">NA</span>
  
  <span class="co">#loop to decide whether to keep each following SNP</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span><span class="op">{</span>
      <span class="co">#if chrom only has 1 SNP, do nothing</span>
    <span class="op">}</span> <span class="kw">else</span><span class="op">{</span>

      <span class="co">#else, use a for loop to determine which SNPs to keep that satisfy our distance criteria</span>
      <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>

        <span class="co">#store distance in base pairs of the current SNP from the previous SNP</span>
        <span class="va">k</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">-</span> <span class="va">prev</span>

        <span class="co">#update the current SNP to be the previous SNP for the next iteration of this loop</span>
        <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
        
      <span class="co">#close for loop</span>
      <span class="op">}</span>
      <span class="co">#k[1]&lt;- fix.sub[2] - fix.sub[1]</span>
    <span class="co">#close else statement</span>
    <span class="op">}</span>

  <span class="co">#make a dataframe with the precise info for each SNP for this chromosome</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>CHROM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">t</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span>, POS<span class="op">=</span><span class="va">fix.sub</span>, dist<span class="op">=</span><span class="va">k</span><span class="op">)</span>

  <span class="co">#now we rbind in the information for this chromosome to the overall df</span>
  <span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">keep.df</span>,<span class="va">chrom.df</span><span class="op">)</span>

  <span class="co">#empty df for this chrom to prepare for the next one</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="cn">NULL</span>

<span class="op">}</span> <span class="co">#close for loop, start over on next chromosome</span>

<span class="co">#order the dataframe to match the order of the input vcf file</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">keep.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">keep.df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">keep.df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">order.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span><span class="op">)</span>,<span class="op">]</span>

<span class="co">#show the lower distribution of SNP distances to see if any fall below the 100bp threshold</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span>, xlim <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1000</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">50000</span>, main <span class="op">=</span> <span class="st">"thinning using SNPfiltR"</span><span class="op">)</span></code></pre></div>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR-method.html">head</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">)</span> <span class="co">#show the tail of SNP distances</span></code></pre></div>
<pre><code>##             CHROM      POS dist
## 5896  Pseudo3_dna 37399303  101
## 1556 Pseudo10_dna 16674460  102
## 5856  Pseudo3_dna 34435555  103
## 4399  Pseudo2_dna 84817274  104
## 3571  Pseudo2_dna 11106890  104
## 3774  Pseudo2_dna 28178166  105</code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">### repeat for the VCFtools filtered vcf</span>
<span class="co">##</span>
<span class="co">#</span>

<span class="co">#generate dataframe containing information for chromosome and bp locality of each SNP</span>
<span class="va">df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#generate list of all unique chromosomes in alphabetical order</span>
<span class="va">chroms</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span><span class="op">)</span><span class="op">)</span>
<span class="co">#intialize empty df to hold filtering</span>
<span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#loop over each chromosome</span>
<span class="co">#for t in vector containing the name of each chromosome</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">t</span> <span class="kw">in</span> <span class="va">chroms</span><span class="op">)</span><span class="op">{</span>

  <span class="co">#isolate the SNP positions on the given chromosome</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span> <span class="op">==</span> <span class="va">t</span><span class="op">]</span><span class="op">)</span>

  <span class="co">#order the positions numerically</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">]</span>

  <span class="co">#set the first position</span>
  <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

  <span class="co">#initialize empty vector</span>
  <span class="va">k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">k</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="cn">NA</span>
  
  <span class="co">#loop to decide whether to keep each following SNP</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span><span class="op">{</span>
      <span class="co">#if chrom only has 1 SNP, do nothing</span>
    <span class="op">}</span> <span class="kw">else</span><span class="op">{</span>

      <span class="co">#else, use a for loop to determine which SNPs to keep that satisfy our distance criteria</span>
      <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>

        <span class="co">#store distance in base pairs of the current SNP from the previous SNP</span>
        <span class="va">k</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">-</span> <span class="va">prev</span>

        <span class="co">#update the current SNP to be the previous SNP for the next iteration of this loop</span>
        <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
        
      <span class="co">#close for loop</span>
      <span class="op">}</span>
      <span class="co">#k[1]&lt;- fix.sub[2] - fix.sub[1]</span>
    <span class="co">#close else statement</span>
    <span class="op">}</span>

  <span class="co">#make a dataframe with the precise info for each SNP for this chromosome</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>CHROM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">t</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span>, POS<span class="op">=</span><span class="va">fix.sub</span>, dist<span class="op">=</span><span class="va">k</span><span class="op">)</span>

  <span class="co">#now we rbind in the information for this chromosome to the overall df</span>
  <span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">keep.df</span>,<span class="va">chrom.df</span><span class="op">)</span>

  <span class="co">#empty df for this chrom to prepare for the next one</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="cn">NULL</span>

<span class="op">}</span> <span class="co">#close for loop, start over on next chromosome</span>

<span class="co">#order the dataframe to match the order of the input vcf file</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">keep.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">keep.df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">keep.df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">order.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span><span class="op">)</span>,<span class="op">]</span>

<span class="co">#show the lower distribution of SNP distances</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span>, xlim <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1000</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">50000</span>, main <span class="op">=</span> <span class="st">"thinning using VCFtools"</span><span class="op">)</span></code></pre></div>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-13-2.png" width="700"></p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR-method.html">head</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">)</span> <span class="co">#show the tail of SNP distances</span></code></pre></div>
<pre><code>##             CHROM       POS dist
## 1834 Pseudo12_dna   1042038    1
## 272   Pseudo1_dna  27993527    1
## 5451 Pseudo24_dna   3659449    1
## 8470  Pseudo5_dna  49662826    1
## 7121  Pseudo4_dna  19499381    1
## 1080  Pseudo1_dna 106078918    1</code></pre>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#putting the dataframe in chromosomal order, we can see right away that SNPs less than 100bp apart have been retained</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">order.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">order.df</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">CHROM</span>, <span class="va">POS</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">order.df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></code></pre></div>
<pre><code>##          CHROM     POS   dist
## 1  Pseudo1_dna   82043     NA
## 2  Pseudo1_dna  763276 681233
## 3  Pseudo1_dna  856505  93229
## 4  Pseudo1_dna  856902    397
## 5  Pseudo1_dna  856957     55
## 6  Pseudo1_dna  857001     44
## 7  Pseudo1_dna  944932  87931
## 8  Pseudo1_dna 1248096 303164
## 9  Pseudo1_dna 1248145     49
## 10 Pseudo1_dna 1290093  41948</code></pre>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#I don't know why this is happening? Potentially VCFtools is not recognizing the way the chromosomes are named in the vcf or something? Either way, safer to use SNPfiltR distance_thin() which we validated here.</span></code></pre></div>
</div>
</div>
<div id="validate-the-min_mac-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-min_mac-function" class="anchor"></a>Validate the min_mac() function</h1>
<div id="filter-each-vcf-with-snpfiltr-3" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr-3" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on each vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<pre><code>## 53.83% of SNPs fell below a minor allele count of 2 and were removed from the VCF</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb89"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<pre><code>## 53.89% of SNPs fell below a minor allele count of 2 and were removed from the VCF</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<pre><code>## 54.33% of SNPs fell below a minor allele count of 2 and were removed from the VCF</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-14-3.png" width="700"></p>
</div>
<div id="filter-each-with-vcftools-3" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools-3" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb95"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--mac</span> 2 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--mac</span> 2 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--mac</span> 2 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --mac 2
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 4617 out of a possible 10000 Sites
## Run Time = 0.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --mac 2
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 9222 out of a possible 20000 Sites
## Run Time = 2.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --mac 2
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 22837 out of a possible 50000 Sites
## Run Time = 3.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files-3" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files-3" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4617
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 4617
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 4617
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant: 4617
## All variants processed</code></pre>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 9222
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 9222
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 9222
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant: 9222
## All variants processed</code></pre>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 22837
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 22837
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 22837
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant: 22837
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-3" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-3" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 137246</code></pre>
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 272988</code></pre>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 667825</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
