<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Accuracy benchmarking • SNPfiltR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Accuracy benchmarking">
<meta property="og:description" content="SNPfiltR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/accuracy-benchmarking.html">Accuracy benchmarking</a>
    </li>
    <li>
      <a href="../articles/performance-benchmarking.html">Performance benchmarking</a>
    </li>
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-RADseq-vignette.html">scrub-jay-RADseq-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-UCE-vignette.html">scrub-jay-UCE-vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="accuracy-benchmarking_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Accuracy benchmarking</h1>
                        <h4 class="author">Devon DeRaad</h4>
            
            <h4 class="date">2/28/2022</h4>
      
      
      <div class="hidden name"><code>accuracy-benchmarking.Rmd</code></div>

    </div>

    
    
<p>This vignette is designed to validate the accuracy of the key functions from the SNPfiltR package against the established VCFtools program. For each function, I filtered 3 vcf files according to the same parameter thresholds using SNPfiltR and VCFtools, and then tested whether each genotype in the output vcf files matched. All functions matched exactly, except for the distance_thin() function. It seems that the SNPfiltR approach accurately thinned the dataset, while the ‘–thin 100’ flag from VCFtools allowed SNPs to remain in the output dataset despite being within 100 base pairs of nearby SNPs. I am not sure why VCFtools is less accurate on these vcf files I used for testing, and encourage more testing from other users in case this is an idiosyncratic issue with the particular vcf files I used. Meanwhile, I suggest using the SNPfiltR function distance_thin() which is validated clearly here, over the VCFtools approach.</p>
<p>All of the input vcf files necessary to fully reproduce this accuracy comparison can be found <a href="https://github.com/DevonDeRaad/SNPfiltR.benchmarking.data">here</a></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SNPfiltR</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR">vcfR</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
##    *****       ***   vcfR   ***       *****
##    This is vcfR 1.12.0 
##      browseVignettes('vcfR') # Documentation
##      citation('vcfR') # Citation
##    *****       *****      *****       *****</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joshuaulrich/microbenchmark/">microbenchmark</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></code></pre></div>
<div id="validate-the-hard_filter-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-hard_filter-function" class="anchor"></a>Validate the hard_filter() function</h1>
<div id="filter-each-vcf-with-snpfiltr" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on the 10K SNP vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed
## [1] "32.58% of genotypes fall below a read depth of 5 and were converted to NA"
## [1] "1.36% of genotypes fall below a genotype quality of 30 and were converted to NA"</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed
## [1] "30.09% of genotypes fall below a read depth of 5 and were converted to NA"
## [1] "1.43% of genotypes fall below a genotype quality of 30 and were converted to NA"</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed
## [1] "31.18% of genotypes fall below a read depth of 5 and were converted to NA"
## [1] "1.37% of genotypes fall below a genotype quality of 30 and were converted to NA"</code></pre>
</div>
<div id="filter-each-with-vcftools" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --minDP 5
##  --minGQ 30
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 10000 out of a possible 10000 Sites
## Run Time = 2.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --minDP 5
##  --minGQ 30
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 20000 out of a possible 20000 Sites
## Run Time = 2.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --minDP 5
##  --minGQ 30
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 50000 out of a possible 50000 Sites
## Run Time = 5.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 127966</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 282173</code></pre>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 664803</code></pre>
</div>
</div>
<div id="validate-the-missing_by_snp-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-missing_by_snp-function" class="anchor"></a>Validate the missing_by_snp() function</h1>
<div id="filter-each-vcf-with-snpfiltr-1" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr-1" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on each vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,cutoff<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed
## [1] "cutoff is specified, filtered vcfR object will be returned"</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<pre><code>## [1] "85.65% of SNPs fell below a completeness cutoff of 0.5 and were removed from the VCF"</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,cutoff<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed
## [1] "cutoff is specified, filtered vcfR object will be returned"</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-6-2.png" width="700"></p>
<pre><code>## [1] "83.8% of SNPs fell below a completeness cutoff of 0.5 and were removed from the VCF"</code></pre>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,cutoff<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed
## [1] "cutoff is specified, filtered vcfR object will be returned"</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-6-3.png" width="700"></p>
<pre><code>## [1] "84.44% of SNPs fell below a completeness cutoff of 0.5 and were removed from the VCF"</code></pre>
</div>
<div id="filter-each-with-vcftools-1" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools-1" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--max-missing</span> 0.5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--max-missing</span> 0.5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--max-missing</span> 0.5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --max-missing 0.5
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 1435 out of a possible 10000 Sites
## Run Time = 1.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --max-missing 0.5
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 3241 out of a possible 20000 Sites
## Run Time = 2.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --max-missing 0.5
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 7782 out of a possible 50000 Sites
## Run Time = 3.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files-1" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files-1" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 1435
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 1435
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 1435
##   row_num: 0
## 
Processed variant 1000
Processed variant: 1435
## All variants processed</code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 3241
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 3241
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 3241
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant: 3241
## All variants processed</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 7782
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 7782
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 7782
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant: 7782
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-1" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-1" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 122778</code></pre>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 277322</code></pre>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 652933</code></pre>
</div>
</div>
<div id="validate-the-min_mac-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-min_mac-function" class="anchor"></a>Validate the min_mac() function</h1>
<div id="filter-each-vcf-with-snpfiltr-2" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr-2" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on each vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<pre><code>## [1] "59.1% of SNPs fell below a minor allele count of 2 and were removed from the VCF"</code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-10-2.png" width="700"></p>
<pre><code>## [1] "58.26% of SNPs fell below a minor allele count of 2 and were removed from the VCF"</code></pre>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-10-3.png" width="700"></p>
<pre><code>## [1] "57.94% of SNPs fell below a minor allele count of 2 and were removed from the VCF"</code></pre>
</div>
<div id="filter-each-with-vcftools-2" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools-2" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb56"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--mac</span> 2 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--mac</span> 2 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--mac</span> 2 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --mac 2
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 4090 out of a possible 10000 Sites
## Run Time = 1.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --mac 2
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 8348 out of a possible 20000 Sites
## Run Time = 1.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --mac 2
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 21031 out of a possible 50000 Sites
## Run Time = 3.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files-2" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files-2" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4090
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 4090
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 4090
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant: 4090
## All variants processed</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 8348
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 8348
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 8348
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant: 8348
## All variants processed</code></pre>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 21031
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 21031
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 21031
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant: 21031
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-2" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-2" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 107663</code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 231584</code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span> <span class="op">==</span> <span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##   TRUE 
## 554511</code></pre>
</div>
</div>
<div id="validate-the-distance_thin-function" class="section level1">
<h1 class="hasAnchor">
<a href="#validate-the-distance_thin-function" class="anchor"></a>Validate the distance_thin() function</h1>
<div id="filter-each-vcf-with-snpfiltr-3" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-vcf-with-snpfiltr-3" class="anchor"></a>Filter each vcf with SNPfiltR</h3>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#perform filtering on each vcf file and save the result</span>
<span class="va">r.10k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/distance_thin.html">distance_thin</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,min.distance<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |======================================================================| 100%
## [1] "2791 out of 10000 input SNPs were not located within 100 base-pairs of another SNP and were retained despite filtering"</code></pre>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.20k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/distance_thin.html">distance_thin</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,min.distance<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |======================================================================| 100%
## [1] "5618 out of 20000 input SNPs were not located within 100 base-pairs of another SNP and were retained despite filtering"</code></pre>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">r.50k</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/distance_thin.html">distance_thin</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,min.distance<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |======================================================================| 100%
## [1] "14478 out of 50000 input SNPs were not located within 100 base-pairs of another SNP and were retained despite filtering"</code></pre>
</div>
<div id="filter-each-with-vcftools-3" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-each-with-vcftools-3" class="anchor"></a>Filter each with VCFtools</h3>
<div class="sourceCode" id="cb76"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--thin</span> 100 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 10K.filtered</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--thin</span> 100 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 20K.filtered</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="ex">/Users/devder/Downloads/vcftools/src/cpp/vcftools</span> <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--thin</span> 100 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="at">--out</span> 50K.filtered</span></code></pre></div>
<pre><code>## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.10K.vcf.gz
##  --recode-INFO-all
##  --thin 100
##  --out 10K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 2408 out of a possible 10000 Sites
## Run Time = 1.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.20K.vcf.gz
##  --recode-INFO-all
##  --thin 100
##  --out 20K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 4826 out of a possible 20000 Sites
## Run Time = 0.00 seconds
## 
## VCFtools - 0.1.17
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --gzvcf benchmark.50K.vcf.gz
##  --recode-INFO-all
##  --thin 100
##  --out 50K.filtered
##  --recode
## 
## Using zlib version: 1.2.11
## After filtering, kept 100 out of 100 Individuals
## Outputting VCF file...
## After filtering, kept 12541 out of a possible 50000 Sites
## Run Time = 2.00 seconds</code></pre>
</div>
<div id="read-in-vcftools-files-3" class="section level3">
<h3 class="hasAnchor">
<a href="#read-in-vcftools-files-3" class="anchor"></a>Read in VCFtools files</h3>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#read in all the vcftools files</span>
<span class="va">bash.10k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/10k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 2408
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 2408
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 2408
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant: 2408
## All variants processed</code></pre>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.20k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/20k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4826
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 4826
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 4826
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant: 4826
## All variants processed</code></pre>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bash.50k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/50k.filtered.recode.vcf"</span><span class="op">)</span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 12541
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 12541
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 12541
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant: 12541
## All variants processed</code></pre>
</div>
<div id="check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-3" class="section level3">
<h3 class="hasAnchor">
<a href="#check-that-every-cell-in-the-genotpye-matrix-matches-for-each-file-3" class="anchor"></a>check that every cell in the genotpye matrix matches for each file</h3>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#Does every cell in the genotype matrix for each camparable vcf match?</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 2791  101</code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 2408  101</code></pre>
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 5618  101</code></pre>
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 4826  101</code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 14478   101</code></pre>
<div class="sourceCode" id="cb94"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 12541   101</code></pre>
<div class="sourceCode" id="cb96"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#right away we can see that the dimensions of these vcf files don't match, indicating that the results differ.</span>

<span class="co">#Let's calculate distance between SNPs in each vcf by hand to make sure both methods work as promised.</span>

<span class="co">### Start with SNPfiltR thinned vcf:</span>
<span class="co">##</span>
<span class="co">#</span>

<span class="co">#generate dataframe containing information for chromosome and bp locality of each SNP</span>
<span class="va">df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#generate list of all unique chromosomes in alphabetical order</span>
<span class="va">chroms</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span><span class="op">)</span><span class="op">)</span>
<span class="co">#intialize empty df to hold filtering</span>
<span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#loop over each chromosome</span>
<span class="co">#for t in vector containing the name of each chromosome</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">t</span> <span class="kw">in</span> <span class="va">chroms</span><span class="op">)</span><span class="op">{</span>

  <span class="co">#isolate the SNP positions on the given chromosome</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span> <span class="op">==</span> <span class="va">t</span><span class="op">]</span><span class="op">)</span>

  <span class="co">#order the positions numerically</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">]</span>

  <span class="co">#set the first position</span>
  <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

  <span class="co">#initialize empty vector</span>
  <span class="va">k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">k</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="cn">NA</span>
  
  <span class="co">#loop to decide whether to keep each following SNP</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span><span class="op">{</span>
      <span class="co">#if chrom only has 1 SNP, do nothing</span>
    <span class="op">}</span> <span class="kw">else</span><span class="op">{</span>

      <span class="co">#else, use a for loop to determine which SNPs to keep that satisfy our distance criteria</span>
      <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>

        <span class="co">#store distance in base pairs of the current SNP from the previous SNP</span>
        <span class="va">k</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">-</span> <span class="va">prev</span>

        <span class="co">#update the current SNP to be the previous SNP for the next iteration of this loop</span>
        <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
        
      <span class="co">#close for loop</span>
      <span class="op">}</span>
      <span class="co">#k[1]&lt;- fix.sub[2] - fix.sub[1]</span>
    <span class="co">#close else statement</span>
    <span class="op">}</span>

  <span class="co">#make a dataframe with the precise info for each SNP for this chromosome</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>CHROM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">t</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span>, POS<span class="op">=</span><span class="va">fix.sub</span>, dist<span class="op">=</span><span class="va">k</span><span class="op">)</span>

  <span class="co">#now we rbind in the information for this chromosome to the overall df</span>
  <span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">keep.df</span>,<span class="va">chrom.df</span><span class="op">)</span>

  <span class="co">#empty df for this chrom to prepare for the next one</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="cn">NULL</span>

<span class="op">}</span> <span class="co">#close for loop, start over on next chromosome</span>

<span class="co">#order the dataframe to match the order of the input vcf file</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">keep.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">keep.df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">keep.df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">order.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span><span class="op">)</span>,<span class="op">]</span>

<span class="co">#show the lower distribution of SNP distances to see if any fall below the 100bp threshold</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span>, xlim <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1000</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">50000</span>, main <span class="op">=</span> <span class="st">"thinning using SNPfiltR"</span>,
          xlab <span class="op">=</span> <span class="st">"Distance from nearest SNP (base pairs)"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="fl">100</span>, col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></code></pre></div>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR-method.html">head</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">)</span> <span class="co">#show the tail of SNP distances</span></code></pre></div>
<pre><code>##            CHROM     POS dist
## 220 Pseudo10_dna 2854287  101
## 331 Pseudo10_dna 3419169  101
## 333 Pseudo10_dna 3420934  101
## 467 Pseudo10_dna 4224139  101
## 469 Pseudo10_dna 4224491  101
## 532 Pseudo10_dna 4498157  101</code></pre>
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#store df</span>
<span class="va">snpfilt.df</span><span class="op">&lt;-</span><span class="va">order.df</span>

<span class="co">### repeat for the VCFtools filtered vcf</span>
<span class="co">##</span>
<span class="co">#</span>

<span class="co">#generate dataframe containing information for chromosome and bp locality of each SNP</span>
<span class="va">df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span>
<span class="co">#generate list of all unique chromosomes in alphabetical order</span>
<span class="va">chroms</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span><span class="op">)</span><span class="op">)</span>
<span class="co">#intialize empty df to hold filtering</span>
<span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#loop over each chromosome</span>
<span class="co">#for t in vector containing the name of each chromosome</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">t</span> <span class="kw">in</span> <span class="va">chroms</span><span class="op">)</span><span class="op">{</span>

  <span class="co">#isolate the SNP positions on the given chromosome</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">[</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span> <span class="op">==</span> <span class="va">t</span><span class="op">]</span><span class="op">)</span>

  <span class="co">#order the positions numerically</span>
  <span class="va">fix.sub</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">]</span>

  <span class="co">#set the first position</span>
  <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>

  <span class="co">#initialize empty vector</span>
  <span class="va">k</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
  <span class="va">k</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="cn">NA</span>
  
  <span class="co">#loop to decide whether to keep each following SNP</span>
    <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">2</span><span class="op">)</span><span class="op">{</span>
      <span class="co">#if chrom only has 1 SNP, do nothing</span>
    <span class="op">}</span> <span class="kw">else</span><span class="op">{</span>

      <span class="co">#else, use a for loop to determine which SNPs to keep that satisfy our distance criteria</span>
      <span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>

        <span class="co">#store distance in base pairs of the current SNP from the previous SNP</span>
        <span class="va">k</span><span class="op">[</span><span class="va">i</span><span class="op">]</span><span class="op">&lt;-</span> <span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">-</span> <span class="va">prev</span>

        <span class="co">#update the current SNP to be the previous SNP for the next iteration of this loop</span>
        <span class="va">prev</span><span class="op">&lt;-</span><span class="va">fix.sub</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
        
      <span class="co">#close for loop</span>
      <span class="op">}</span>
      <span class="co">#k[1]&lt;- fix.sub[2] - fix.sub[1]</span>
    <span class="co">#close else statement</span>
    <span class="op">}</span>

  <span class="co">#make a dataframe with the precise info for each SNP for this chromosome</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>CHROM<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="va">t</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">fix.sub</span><span class="op">)</span><span class="op">)</span>, POS<span class="op">=</span><span class="va">fix.sub</span>, dist<span class="op">=</span><span class="va">k</span><span class="op">)</span>

  <span class="co">#now we rbind in the information for this chromosome to the overall df</span>
  <span class="va">keep.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">keep.df</span>,<span class="va">chrom.df</span><span class="op">)</span>

  <span class="co">#empty df for this chrom to prepare for the next one</span>
  <span class="va">chrom.df</span><span class="op">&lt;-</span><span class="cn">NULL</span>

<span class="op">}</span> <span class="co">#close for loop, start over on next chromosome</span>

<span class="co">#order the dataframe to match the order of the input vcf file</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">keep.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">keep.df</span><span class="op">$</span><span class="va">CHROM</span>,<span class="va">keep.df</span><span class="op">$</span><span class="va">POS</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">order.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span><span class="op">)</span>,<span class="op">]</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">$</span><span class="va">dist</span>, xlim <span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">1000</span><span class="op">)</span>, breaks <span class="op">=</span> <span class="fl">50000</span>, main <span class="op">=</span> <span class="st">"thinning using VCFtools"</span>,
     xlab <span class="op">=</span> <span class="st">"Distance from nearest SNP (base pairs)"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="fl">100</span>, col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></code></pre></div>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR-method.html">head</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">)</span> <span class="co">#show the tail of SNP distances</span></code></pre></div>
<pre><code>##             CHROM      POS dist
## 2197 Pseudo10_dna 16289157  100
## 464  Pseudo10_dna  4498157  101
## 2257 Pseudo10_dna 16585988  101
## 163  Pseudo10_dna  2575297  102
## 1834 Pseudo10_dna 13584214  102
## 848  Pseudo10_dna  6642570  103</code></pre>
<div class="sourceCode" id="cb102"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#putting the dataframe in chromosomal order, we can see right away that SNPs less than 100bp apart have been retained</span>
<span class="va">order.df</span><span class="op">&lt;-</span><span class="va">order.df</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="va">order.df</span>, <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">CHROM</span>, <span class="va">POS</span><span class="op">)</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">order.df</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span></code></pre></div>
<pre><code>##           CHROM    POS  dist
## 1  Pseudo10_dna  25891    NA
## 2  Pseudo10_dna  31885  5994
## 3  Pseudo10_dna  85182 53297
## 4  Pseudo10_dna 135731 50549
## 5  Pseudo10_dna 156488 20757
## 6  Pseudo10_dna 163096  6608
## 7  Pseudo10_dna 178396 15300
## 8  Pseudo10_dna 185542  7146
## 9  Pseudo10_dna 225731 40189
## 10 Pseudo10_dna 228849  3118</code></pre>
<div class="sourceCode" id="cb104"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plot the two together on the same histogram</span>
<span class="va">order.df</span><span class="op">$</span><span class="va">approach</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"VCFtools"</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">order.df</span><span class="op">)</span><span class="op">)</span>
<span class="va">snpfilt.df</span><span class="op">$</span><span class="va">approach</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"SNPfiltR"</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">snpfilt.df</span><span class="op">)</span><span class="op">)</span>
<span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span><span class="op">(</span><span class="va">order.df</span>,<span class="va">snpfilt.df</span><span class="op">)</span>

<span class="co">#plot both</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">sum.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">dist</span>, color<span class="op">=</span><span class="va">approach</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>bins<span class="op">=</span><span class="fl">100</span>, fill<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1000</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span>label <span class="op">=</span> <span class="st">"Distance from nearest SNP (base pairs)"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">100</span>, linetype<span class="op">=</span><span class="st">"dashed"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 3756 rows containing non-finite values (stat_bin).</code></pre>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-17-3.png" width="700"></p>
<div class="sourceCode" id="cb107"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">hists</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">sum.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">dist</span>, color<span class="op">=</span><span class="va">approach</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>bins<span class="op">=</span><span class="fl">100</span>, fill<span class="op">=</span><span class="st">"white"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">2500</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span>label <span class="op">=</span> <span class="st">"Distance from nearest SNP (base pairs)"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_vline</a></span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">100</span>, linetype<span class="op">=</span><span class="st">"dashed"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.8</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span>

<span class="co">#save plot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">hists</span>,
      filename <span class="op">=</span> <span class="st">"~/Desktop/SNPfiltR.mol.ecol.resour.submission/resubmission/validate.thinning.pdf"</span>,
      height <span class="op">=</span> <span class="fl">2</span>, width <span class="op">=</span> <span class="fl">8</span>, units <span class="op">=</span> <span class="st">"in"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 3061 rows containing non-finite values (stat_bin).

## Warning: Removed 4 rows containing missing values (geom_bar).</code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#see how many retained SNPs overlap between the two approaches for each file</span>
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  1212  1579</code></pre>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.20k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  2444  3174</code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.50k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  6170  8308</code></pre>
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#store number of SNPs for each approach as a vector</span>
<span class="va">r.snps</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">r.snps</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>
<span class="va">r.snps</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>
<span class="va">r.snps</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>

<span class="co">#store number of bash SNPs</span>
<span class="va">bash.snps</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">bash.snps</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>
<span class="va">bash.snps</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">bash.20k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>
<span class="va">bash.snps</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">bash.50k</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>

<span class="va">bar.frame</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>snps<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">r.snps</span>,<span class="va">bash.snps</span><span class="op">)</span>,
                      approach<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"SNPfiltR::distance_thin()"</span>,<span class="st">"SNPfiltR::distance_thin()"</span>,<span class="st">"SNPfiltR::distance_thin()"</span>,
                               <span class="st">"VCFtools --thin"</span>,<span class="st">"VCFtools --thin"</span>,<span class="st">"VCFtools --thin"</span><span class="op">)</span>,
                      input.snps<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"10K"</span>,<span class="st">"20K"</span>,<span class="st">"50K"</span>,
                                   <span class="st">"10K"</span>,<span class="st">"20K"</span>,<span class="st">"50K"</span><span class="op">)</span>
                      <span class="op">)</span>

<span class="co">## Grouped bar plot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">bar.frame</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">approach</span>, y<span class="op">=</span><span class="va">snps</span>, x<span class="op">=</span><span class="va">input.snps</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"SNPs in input vcf file"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Number of thinned SNPs"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-17-4.png" width="700"></p>
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">thin.bar</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">bar.frame</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">approach</span>, y<span class="op">=</span><span class="va">snps</span>, x<span class="op">=</span><span class="va">input.snps</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>position<span class="op">=</span><span class="st">"dodge"</span>, stat<span class="op">=</span><span class="st">"identity"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"SNPs in input vcf file"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Number of thinned SNPs"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span>

<span class="co">#save plot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">thin.bar</span>,
      filename <span class="op">=</span> <span class="st">"~/Desktop/SNPfiltR.mol.ecol.resour.submission/resubmission/thinning.comp.pdf"</span>,
      height <span class="op">=</span> <span class="fl">3</span>, width <span class="op">=</span> <span class="fl">4.5</span>, units <span class="op">=</span> <span class="st">"in"</span><span class="op">)</span></code></pre></div>
</div>
<div id="make-final-table-showing-genotpye-concordance-across-comparisons" class="section level3">
<h3 class="hasAnchor">
<a href="#make-final-table-showing-genotpye-concordance-across-comparisons" class="anchor"></a>Make final table showing genotpye concordance across comparisons</h3>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># print table showing concordance</span>
<span class="va">concord</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>concordance<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,
                                  <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.10k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span><span class="op">)</span>,
                                  <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.20k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.20k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span><span class="op">)</span>,
                                  <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html">with</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">bash.50k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">CHROM</span>,<span class="va">POS</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">r.50k</span><span class="op">@</span><span class="va">fix</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
                      filtering<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"GQ"</span>,<span class="st">"GQ"</span>,<span class="st">"GQ"</span>,<span class="st">"missing data"</span>,<span class="st">"missing data"</span>,<span class="st">"missing data"</span>,
                                  <span class="st">"MAC"</span>,<span class="st">"MAC"</span>,<span class="st">"MAC"</span>,<span class="st">"distance thin"</span>,<span class="st">"distance thin"</span>,<span class="st">"distance thin"</span><span class="op">)</span>,
                      input.snps<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"10K"</span>,<span class="st">"20K"</span>,<span class="st">"50K"</span>,
                                   <span class="st">"10K"</span>,<span class="st">"20K"</span>,<span class="st">"50K"</span>,
                                   <span class="st">"10K"</span>,<span class="st">"20K"</span>,<span class="st">"50K"</span>,
                                   <span class="st">"10K"</span>,<span class="st">"20K"</span>,<span class="st">"50K"</span><span class="op">)</span>
                      <span class="op">)</span>

<span class="co">#print table</span>
<span class="va">heat</span> <span class="op">&lt;-</span> <span class="fu">reshape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape/man/melt-24.html">melt</a></span><span class="op">(</span><span class="va">concord</span><span class="op">)</span></code></pre></div>
<pre><code>## Using filtering, input.snps as id variables</code></pre>
<div class="sourceCode" id="cb119"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">heat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">input.snps</span>, y<span class="op">=</span><span class="va">filtering</span>, fill<span class="op">=</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html">geom_tile</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"white"</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">heat</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">value</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_fill_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"white"</span>, high <span class="op">=</span> <span class="st">"cornflowerblue"</span>, space <span class="op">=</span> <span class="st">"Lab"</span>, name<span class="op">=</span><span class="st">"genotype\nconcordance"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"type of filtering"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Number of SNPs in input vcf file"</span><span class="op">)</span></code></pre></div>
<p><img src="accuracy-benchmarking_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">geno.concord.plot</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">heat</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">input.snps</span>, y<span class="op">=</span><span class="va">filtering</span>, fill<span class="op">=</span><span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_tile.html">geom_tile</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"white"</span>, size<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data<span class="op">=</span><span class="va">heat</span>,<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>label<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">value</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_minimal</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_fill_gradient2</a></span><span class="op">(</span>low <span class="op">=</span> <span class="st">"white"</span>, high <span class="op">=</span> <span class="st">"cornflowerblue"</span>, space <span class="op">=</span> <span class="st">"Lab"</span>, name<span class="op">=</span><span class="st">"genotype\nconcordance"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"type of filtering"</span><span class="op">)</span><span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"Number of SNPs in input vcf file"</span><span class="op">)</span>

<span class="co">#save plot</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">geno.concord.plot</span>,
      filename <span class="op">=</span> <span class="st">"~/Desktop/SNPfiltR.mol.ecol.resour.submission/resubmission/geno.concord.pdf"</span>,
      height <span class="op">=</span> <span class="fl">3</span>, width <span class="op">=</span> <span class="fl">5</span>, units <span class="op">=</span> <span class="st">"in"</span><span class="op">)</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
