<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Performance benchmarking • SNPfiltR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Performance benchmarking">
<meta property="og:description" content="SNPfiltR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/accuracy-benchmarking.html">accuracy-benchmarking</a>
    </li>
    <li>
      <a href="../articles/performance-benchmarking.html">Performance benchmarking</a>
    </li>
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-RADseq-vignette.html">scrub-jay-RADseq-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-UCE-vignette.html">scrub-jay-UCE-vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Performance benchmarking</h1>
                        <h4 data-toc-skip class="author">Devon
DeRaad</h4>
            
            <h4 data-toc-skip class="date">10/27/2021</h4>
      
      
      <div class="hidden name"><code>performance-benchmarking.Rmd</code></div>

    </div>

    
    
<p>This vignette shows the relative filtering time of the hard_filter()
function from SNPfiltR versus the same filtering approach using
VCFtools. I repeated this procedure and found little difference between
compressed (.gz) vcf input and standard vcf input, so the comparisons
shown here are done with gzipped vcf files as input. All of the input
vcf files necessary to fully reproduce this performance comparison can
be found <a href="https://github.com/DevonDeRaad/SNPfiltR.benchmarking.data" class="external-link">here</a></p>
<div class="section level4">
<h4 id="we-are-going-to-do-performance-benchmarking-on-the-hard_filter-function-from-the-snpfiltr-package-against-the-same-filtering-approach-implemented-in-vcftools--i-have-already-generated-vcf-files-with-500k-400k-300k-200k-100k-50k-20k-and-10k-snps-each-with-100-individuals-in-order-to-compare-performance-across-file-sizes--for-each-vcf-file-we-will-use-the-r-package-microbenchmark-to-benchmark-the-time-it-takes-to">We are going to do performance benchmarking on the hard_filter()
function from the SNPfiltR package, against the same filtering approach
implemented in VCFtools. I have already generated vcf files with 500K,
400K, 300K, 200K, 100K, 50K, 20K, and 10K SNPs, (each with 100
individuals) in order to compare performance across file sizes. For each
vcf file, we will use the R package microbenchmark to benchmark the time
it takes to:<a class="anchor" aria-label="anchor" href="#we-are-going-to-do-performance-benchmarking-on-the-hard_filter-function-from-the-snpfiltr-package-against-the-same-filtering-approach-implemented-in-vcftools--i-have-already-generated-vcf-files-with-500k-400k-300k-200k-100k-50k-20k-and-10k-snps-each-with-100-individuals-in-order-to-compare-performance-across-file-sizes--for-each-vcf-file-we-will-use-the-r-package-microbenchmark-to-benchmark-the-time-it-takes-to"></a>
</h4>
<div class="section level5">
<h5 id="read-in-the-vcf-using-vcfrread-vcfr-and-use-snpfiltrhard_filter-to-filter-the-file-to-a-minimum-depth-of-5-per-genotype-and-a-minimum-quality-of-30-per-genotype-">1) Read in the vcf using vcfR::read.vcfR() and use
SNPfiltR::hard_filter() to filter the file to a minimum depth of 5 per
genotype and a minimum quality of 30 per genotype.<a class="anchor" aria-label="anchor" href="#read-in-the-vcf-using-vcfrread-vcfr-and-use-snpfiltrhard_filter-to-filter-the-file-to-a-minimum-depth-of-5-per-genotype-and-a-minimum-quality-of-30-per-genotype-"></a>
</h5>
</div>
<div class="section level5">
<h5 id="use-snpfiltrhard_filter-to-filter-the-file-to-a-minimum-depth-of-5-per-genotype-and-a-minimum-quality-of-30-per-genotype-with-the-vcfr-object-already-read-in-">2) Use SNPfiltR::hard_filter() to filter the file to a minimum depth
of 5 per genotype and a minimum quality of 30 per genotype, with the
vcfR object already read in.<a class="anchor" aria-label="anchor" href="#use-snpfiltrhard_filter-to-filter-the-file-to-a-minimum-depth-of-5-per-genotype-and-a-minimum-quality-of-30-per-genotype-with-the-vcfr-object-already-read-in-"></a>
</h5>
</div>
</div>
<div class="section level4">
<h4 id="using-microbenchmark-i-will-execute-three-replicates-for-each-approach-and-record-the-mean-value-of-the-three-replicates--then-i-will-execute-the-same-filter-three-times-minimum-depth-5-minimum-gq-30-using-vcftools-and-record-the-mean-of-the-three-replicates-">Using microbenchmark, I will execute three replicates for each
approach, and record the mean value of the three replicates. Then I will
execute the same filter three times (minimum depth = 5, minimum gq =30)
using VCFtools, and record the mean of the three replicates.<a class="anchor" aria-label="anchor" href="#using-microbenchmark-i-will-execute-three-replicates-for-each-approach-and-record-the-mean-value-of-the-three-replicates--then-i-will-execute-the-same-filter-three-times-minimum-depth-5-minimum-gq-30-using-vcftools-and-record-the-mean-of-the-three-replicates-"></a>
</h4>
</div>
<div class="section level4">
<h4 id="after-doing-this-for-each-vcf-file-we-should-get-a-sense-of-the-performance-of-the-snpfiltr-package-relative-to-the-state-of-the-art-program-vcftools-and-how-each-program-scales-with-input-file-size-">After doing this for each vcf file, we should get a sense of the
performance of the SNPfiltR package relative to the state of the art
program VCFtools, and how each program scales with input file size.<a class="anchor" aria-label="anchor" href="#after-doing-this-for-each-vcf-file-we-should-get-a-sense-of-the-performance-of-the-snpfiltr-package-relative-to-the-state-of-the-art-program-vcftools-and-how-each-program-scales-with-input-file-size-"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SNPfiltR</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## This is SNPfiltR v.1.0.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Detailed usage information is available at: devonderaad.github.io/SNPfiltR/ </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use SNPfiltR in your published work, please cite the following papers: </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## DeRaad, D.A. (2022), SNPfiltR: an R package for interactive and reproducible SNP filtering. Molecular Ecology Resources, 00, 1-15. http://doi.org/10.1111/1755-0998.13618 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Knaus, Brian J., and Niklaus J. Grunwald. 2017. VCFR: a package to manipulate and visualize variant call format data in R. Molecular Ecology Resources, 17.1:44-53. http://doi.org/10.1111/1755-0998.12549</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR" class="external-link">vcfR</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##    *****       ***   vcfR   ***       *****</span></span>
<span><span class="co">##    This is vcfR 1.14.0 </span></span>
<span><span class="co">##      browseVignettes('vcfR') # Documentation</span></span>
<span><span class="co">##      citation('vcfR') # Citation</span></span>
<span><span class="co">##    *****       *****      *****       *****</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/joshuaulrich/microbenchmark/" class="external-link">microbenchmark</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://lubridate.tidyverse.org" class="external-link">lubridate</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'lubridate'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     date, intersect, setdiff, union</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="benchmark-10k">Benchmark 10K<a class="anchor" aria-label="anchor" href="#benchmark-10k"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#10K</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 10K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 32.58% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 32.58% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 32.58% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 10K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.10K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 10000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 10000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 10000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant: 10000
## All variants processed</code></pre>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 20K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 32.58% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 32.58% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 32.58% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="benchmark-20k">Benchmark 20K<a class="anchor" aria-label="anchor" href="#benchmark-20k"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#20K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 20K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 30.09% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.43% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 30.09% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.43% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 30.09% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.43% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 20K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.20K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 20000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 20000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 20000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant: 20000
## All variants processed</code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 20K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 30.09% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.43% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 30.09% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.43% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 30.09% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.43% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="benchmark-50k">Benchmark 50K<a class="anchor" aria-label="anchor" href="#benchmark-50k"></a>
</h2>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#50K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 50K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 31.18% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.37% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 31.18% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.37% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 31.18% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.37% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 50K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.50K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 50000
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 50000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 50000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant: 50000
## All variants processed</code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 50K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 31.18% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.37% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 31.18% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.37% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 31.18% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.37% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="benchmark-100k">Benchmark 100K<a class="anchor" aria-label="anchor" href="#benchmark-100k"></a>
</h2>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#100K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 100K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.100K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 1e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 100000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 100000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant: 100000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 30.89% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 1e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 100000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 100000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant: 100000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 30.89% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 1e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 100000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 100000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant: 100000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 30.89% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 100K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.100K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 1e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 100000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 100000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant: 100000
## All variants processed</code></pre>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 100K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 30.89% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 30.89% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 30.89% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.36% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="benchmark-200k">Benchmark 200K<a class="anchor" aria-label="anchor" href="#benchmark-200k"></a>
</h2>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#200K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 200K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.200K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 2e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 200000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 200000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant: 200000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.5% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.28% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 2e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 200000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 200000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant: 200000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.5% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.28% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 2e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 200000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 200000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant: 200000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.5% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.28% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 200K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.200K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 2e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 200000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 200000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant: 200000
## All variants processed</code></pre>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 200K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 29.5% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.28% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.5% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.28% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.5% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.28% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb92"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="benchmark-300k">Benchmark 300K<a class="anchor" aria-label="anchor" href="#benchmark-300k"></a>
</h2>
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#300K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 300K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.300K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 3e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 300000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 300000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant: 300000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 3e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 300000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 300000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant: 300000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 3e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 300000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 300000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant: 300000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb101"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 300K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.300K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 3e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 300000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 300000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant: 300000
## All variants processed</code></pre>
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 300K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.29% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
<p>#Benchmark 400K</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#400K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 400K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.400K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 400000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 400000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant: 400000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.2% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 400000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 400000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant: 400000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.2% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 400000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 400000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant: 400000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.2% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb118"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 400K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.400K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 4e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 400000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 400000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant: 400000
## All variants processed</code></pre>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 400K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 29.2% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.2% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.2% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.31% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb126"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="benchmark-500k">Benchmark 500K<a class="anchor" aria-label="anchor" href="#benchmark-500k"></a>
</h2>
<div class="sourceCode" id="cb127"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#500K</span></span>
<span><span class="co">#remove objects currently in the working directory to avoid internal memory issues slowing down the functions and affecting benchmarking accuracy</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#now benchmark with having to read in the file itself</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 500K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.500K.vcf.gz"</span><span class="op">)</span>,depth <span class="op">=</span> <span class="fl">5</span>,gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 5e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 500000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 500000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant 401000
Processed variant 402000
Processed variant 403000
Processed variant 404000
Processed variant 405000
Processed variant 406000
Processed variant 407000
Processed variant 408000
Processed variant 409000
Processed variant 410000
Processed variant 411000
Processed variant 412000
Processed variant 413000
Processed variant 414000
Processed variant 415000
Processed variant 416000
Processed variant 417000
Processed variant 418000
Processed variant 419000
Processed variant 420000
Processed variant 421000
Processed variant 422000
Processed variant 423000
Processed variant 424000
Processed variant 425000
Processed variant 426000
Processed variant 427000
Processed variant 428000
Processed variant 429000
Processed variant 430000
Processed variant 431000
Processed variant 432000
Processed variant 433000
Processed variant 434000
Processed variant 435000
Processed variant 436000
Processed variant 437000
Processed variant 438000
Processed variant 439000
Processed variant 440000
Processed variant 441000
Processed variant 442000
Processed variant 443000
Processed variant 444000
Processed variant 445000
Processed variant 446000
Processed variant 447000
Processed variant 448000
Processed variant 449000
Processed variant 450000
Processed variant 451000
Processed variant 452000
Processed variant 453000
Processed variant 454000
Processed variant 455000
Processed variant 456000
Processed variant 457000
Processed variant 458000
Processed variant 459000
Processed variant 460000
Processed variant 461000
Processed variant 462000
Processed variant 463000
Processed variant 464000
Processed variant 465000
Processed variant 466000
Processed variant 467000
Processed variant 468000
Processed variant 469000
Processed variant 470000
Processed variant 471000
Processed variant 472000
Processed variant 473000
Processed variant 474000
Processed variant 475000
Processed variant 476000
Processed variant 477000
Processed variant 478000
Processed variant 479000
Processed variant 480000
Processed variant 481000
Processed variant 482000
Processed variant 483000
Processed variant 484000
Processed variant 485000
Processed variant 486000
Processed variant 487000
Processed variant 488000
Processed variant 489000
Processed variant 490000
Processed variant 491000
Processed variant 492000
Processed variant 493000
Processed variant 494000
Processed variant 495000
Processed variant 496000
Processed variant 497000
Processed variant 498000
Processed variant 499000
Processed variant 500000
Processed variant: 500000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.32% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 5e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 500000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 500000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant 401000
Processed variant 402000
Processed variant 403000
Processed variant 404000
Processed variant 405000
Processed variant 406000
Processed variant 407000
Processed variant 408000
Processed variant 409000
Processed variant 410000
Processed variant 411000
Processed variant 412000
Processed variant 413000
Processed variant 414000
Processed variant 415000
Processed variant 416000
Processed variant 417000
Processed variant 418000
Processed variant 419000
Processed variant 420000
Processed variant 421000
Processed variant 422000
Processed variant 423000
Processed variant 424000
Processed variant 425000
Processed variant 426000
Processed variant 427000
Processed variant 428000
Processed variant 429000
Processed variant 430000
Processed variant 431000
Processed variant 432000
Processed variant 433000
Processed variant 434000
Processed variant 435000
Processed variant 436000
Processed variant 437000
Processed variant 438000
Processed variant 439000
Processed variant 440000
Processed variant 441000
Processed variant 442000
Processed variant 443000
Processed variant 444000
Processed variant 445000
Processed variant 446000
Processed variant 447000
Processed variant 448000
Processed variant 449000
Processed variant 450000
Processed variant 451000
Processed variant 452000
Processed variant 453000
Processed variant 454000
Processed variant 455000
Processed variant 456000
Processed variant 457000
Processed variant 458000
Processed variant 459000
Processed variant 460000
Processed variant 461000
Processed variant 462000
Processed variant 463000
Processed variant 464000
Processed variant 465000
Processed variant 466000
Processed variant 467000
Processed variant 468000
Processed variant 469000
Processed variant 470000
Processed variant 471000
Processed variant 472000
Processed variant 473000
Processed variant 474000
Processed variant 475000
Processed variant 476000
Processed variant 477000
Processed variant 478000
Processed variant 479000
Processed variant 480000
Processed variant 481000
Processed variant 482000
Processed variant 483000
Processed variant 484000
Processed variant 485000
Processed variant 486000
Processed variant 487000
Processed variant 488000
Processed variant 489000
Processed variant 490000
Processed variant 491000
Processed variant 492000
Processed variant 493000
Processed variant 494000
Processed variant 495000
Processed variant 496000
Processed variant 497000
Processed variant 498000
Processed variant 499000
Processed variant 500000
Processed variant: 500000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.32% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 5e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 500000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 500000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant 401000
Processed variant 402000
Processed variant 403000
Processed variant 404000
Processed variant 405000
Processed variant 406000
Processed variant 407000
Processed variant 408000
Processed variant 409000
Processed variant 410000
Processed variant 411000
Processed variant 412000
Processed variant 413000
Processed variant 414000
Processed variant 415000
Processed variant 416000
Processed variant 417000
Processed variant 418000
Processed variant 419000
Processed variant 420000
Processed variant 421000
Processed variant 422000
Processed variant 423000
Processed variant 424000
Processed variant 425000
Processed variant 426000
Processed variant 427000
Processed variant 428000
Processed variant 429000
Processed variant 430000
Processed variant 431000
Processed variant 432000
Processed variant 433000
Processed variant 434000
Processed variant 435000
Processed variant 436000
Processed variant 437000
Processed variant 438000
Processed variant 439000
Processed variant 440000
Processed variant 441000
Processed variant 442000
Processed variant 443000
Processed variant 444000
Processed variant 445000
Processed variant 446000
Processed variant 447000
Processed variant 448000
Processed variant 449000
Processed variant 450000
Processed variant 451000
Processed variant 452000
Processed variant 453000
Processed variant 454000
Processed variant 455000
Processed variant 456000
Processed variant 457000
Processed variant 458000
Processed variant 459000
Processed variant 460000
Processed variant 461000
Processed variant 462000
Processed variant 463000
Processed variant 464000
Processed variant 465000
Processed variant 466000
Processed variant 467000
Processed variant 468000
Processed variant 469000
Processed variant 470000
Processed variant 471000
Processed variant 472000
Processed variant 473000
Processed variant 474000
Processed variant 475000
Processed variant 476000
Processed variant 477000
Processed variant 478000
Processed variant 479000
Processed variant 480000
Processed variant 481000
Processed variant 482000
Processed variant 483000
Processed variant 484000
Processed variant 485000
Processed variant 486000
Processed variant 487000
Processed variant 488000
Processed variant 489000
Processed variant 490000
Processed variant 491000
Processed variant 492000
Processed variant 493000
Processed variant 494000
Processed variant 495000
Processed variant 496000
Processed variant 497000
Processed variant 498000
Processed variant 499000
Processed variant 500000
Processed variant: 500000
## All variants processed</code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.32% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb135"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in 500K benchmarking vcf</span></span>
<span><span class="va">x</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.500K.vcf.gz"</span><span class="op">)</span></span></code></pre></div>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 5e+05
##   column count: 109
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 500000
##   Character matrix gt cols: 109
##   skip: 0
##   nrows: 500000
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant 6000
Processed variant 7000
Processed variant 8000
Processed variant 9000
Processed variant 10000
Processed variant 11000
Processed variant 12000
Processed variant 13000
Processed variant 14000
Processed variant 15000
Processed variant 16000
Processed variant 17000
Processed variant 18000
Processed variant 19000
Processed variant 20000
Processed variant 21000
Processed variant 22000
Processed variant 23000
Processed variant 24000
Processed variant 25000
Processed variant 26000
Processed variant 27000
Processed variant 28000
Processed variant 29000
Processed variant 30000
Processed variant 31000
Processed variant 32000
Processed variant 33000
Processed variant 34000
Processed variant 35000
Processed variant 36000
Processed variant 37000
Processed variant 38000
Processed variant 39000
Processed variant 40000
Processed variant 41000
Processed variant 42000
Processed variant 43000
Processed variant 44000
Processed variant 45000
Processed variant 46000
Processed variant 47000
Processed variant 48000
Processed variant 49000
Processed variant 50000
Processed variant 51000
Processed variant 52000
Processed variant 53000
Processed variant 54000
Processed variant 55000
Processed variant 56000
Processed variant 57000
Processed variant 58000
Processed variant 59000
Processed variant 60000
Processed variant 61000
Processed variant 62000
Processed variant 63000
Processed variant 64000
Processed variant 65000
Processed variant 66000
Processed variant 67000
Processed variant 68000
Processed variant 69000
Processed variant 70000
Processed variant 71000
Processed variant 72000
Processed variant 73000
Processed variant 74000
Processed variant 75000
Processed variant 76000
Processed variant 77000
Processed variant 78000
Processed variant 79000
Processed variant 80000
Processed variant 81000
Processed variant 82000
Processed variant 83000
Processed variant 84000
Processed variant 85000
Processed variant 86000
Processed variant 87000
Processed variant 88000
Processed variant 89000
Processed variant 90000
Processed variant 91000
Processed variant 92000
Processed variant 93000
Processed variant 94000
Processed variant 95000
Processed variant 96000
Processed variant 97000
Processed variant 98000
Processed variant 99000
Processed variant 100000
Processed variant 101000
Processed variant 102000
Processed variant 103000
Processed variant 104000
Processed variant 105000
Processed variant 106000
Processed variant 107000
Processed variant 108000
Processed variant 109000
Processed variant 110000
Processed variant 111000
Processed variant 112000
Processed variant 113000
Processed variant 114000
Processed variant 115000
Processed variant 116000
Processed variant 117000
Processed variant 118000
Processed variant 119000
Processed variant 120000
Processed variant 121000
Processed variant 122000
Processed variant 123000
Processed variant 124000
Processed variant 125000
Processed variant 126000
Processed variant 127000
Processed variant 128000
Processed variant 129000
Processed variant 130000
Processed variant 131000
Processed variant 132000
Processed variant 133000
Processed variant 134000
Processed variant 135000
Processed variant 136000
Processed variant 137000
Processed variant 138000
Processed variant 139000
Processed variant 140000
Processed variant 141000
Processed variant 142000
Processed variant 143000
Processed variant 144000
Processed variant 145000
Processed variant 146000
Processed variant 147000
Processed variant 148000
Processed variant 149000
Processed variant 150000
Processed variant 151000
Processed variant 152000
Processed variant 153000
Processed variant 154000
Processed variant 155000
Processed variant 156000
Processed variant 157000
Processed variant 158000
Processed variant 159000
Processed variant 160000
Processed variant 161000
Processed variant 162000
Processed variant 163000
Processed variant 164000
Processed variant 165000
Processed variant 166000
Processed variant 167000
Processed variant 168000
Processed variant 169000
Processed variant 170000
Processed variant 171000
Processed variant 172000
Processed variant 173000
Processed variant 174000
Processed variant 175000
Processed variant 176000
Processed variant 177000
Processed variant 178000
Processed variant 179000
Processed variant 180000
Processed variant 181000
Processed variant 182000
Processed variant 183000
Processed variant 184000
Processed variant 185000
Processed variant 186000
Processed variant 187000
Processed variant 188000
Processed variant 189000
Processed variant 190000
Processed variant 191000
Processed variant 192000
Processed variant 193000
Processed variant 194000
Processed variant 195000
Processed variant 196000
Processed variant 197000
Processed variant 198000
Processed variant 199000
Processed variant 200000
Processed variant 201000
Processed variant 202000
Processed variant 203000
Processed variant 204000
Processed variant 205000
Processed variant 206000
Processed variant 207000
Processed variant 208000
Processed variant 209000
Processed variant 210000
Processed variant 211000
Processed variant 212000
Processed variant 213000
Processed variant 214000
Processed variant 215000
Processed variant 216000
Processed variant 217000
Processed variant 218000
Processed variant 219000
Processed variant 220000
Processed variant 221000
Processed variant 222000
Processed variant 223000
Processed variant 224000
Processed variant 225000
Processed variant 226000
Processed variant 227000
Processed variant 228000
Processed variant 229000
Processed variant 230000
Processed variant 231000
Processed variant 232000
Processed variant 233000
Processed variant 234000
Processed variant 235000
Processed variant 236000
Processed variant 237000
Processed variant 238000
Processed variant 239000
Processed variant 240000
Processed variant 241000
Processed variant 242000
Processed variant 243000
Processed variant 244000
Processed variant 245000
Processed variant 246000
Processed variant 247000
Processed variant 248000
Processed variant 249000
Processed variant 250000
Processed variant 251000
Processed variant 252000
Processed variant 253000
Processed variant 254000
Processed variant 255000
Processed variant 256000
Processed variant 257000
Processed variant 258000
Processed variant 259000
Processed variant 260000
Processed variant 261000
Processed variant 262000
Processed variant 263000
Processed variant 264000
Processed variant 265000
Processed variant 266000
Processed variant 267000
Processed variant 268000
Processed variant 269000
Processed variant 270000
Processed variant 271000
Processed variant 272000
Processed variant 273000
Processed variant 274000
Processed variant 275000
Processed variant 276000
Processed variant 277000
Processed variant 278000
Processed variant 279000
Processed variant 280000
Processed variant 281000
Processed variant 282000
Processed variant 283000
Processed variant 284000
Processed variant 285000
Processed variant 286000
Processed variant 287000
Processed variant 288000
Processed variant 289000
Processed variant 290000
Processed variant 291000
Processed variant 292000
Processed variant 293000
Processed variant 294000
Processed variant 295000
Processed variant 296000
Processed variant 297000
Processed variant 298000
Processed variant 299000
Processed variant 300000
Processed variant 301000
Processed variant 302000
Processed variant 303000
Processed variant 304000
Processed variant 305000
Processed variant 306000
Processed variant 307000
Processed variant 308000
Processed variant 309000
Processed variant 310000
Processed variant 311000
Processed variant 312000
Processed variant 313000
Processed variant 314000
Processed variant 315000
Processed variant 316000
Processed variant 317000
Processed variant 318000
Processed variant 319000
Processed variant 320000
Processed variant 321000
Processed variant 322000
Processed variant 323000
Processed variant 324000
Processed variant 325000
Processed variant 326000
Processed variant 327000
Processed variant 328000
Processed variant 329000
Processed variant 330000
Processed variant 331000
Processed variant 332000
Processed variant 333000
Processed variant 334000
Processed variant 335000
Processed variant 336000
Processed variant 337000
Processed variant 338000
Processed variant 339000
Processed variant 340000
Processed variant 341000
Processed variant 342000
Processed variant 343000
Processed variant 344000
Processed variant 345000
Processed variant 346000
Processed variant 347000
Processed variant 348000
Processed variant 349000
Processed variant 350000
Processed variant 351000
Processed variant 352000
Processed variant 353000
Processed variant 354000
Processed variant 355000
Processed variant 356000
Processed variant 357000
Processed variant 358000
Processed variant 359000
Processed variant 360000
Processed variant 361000
Processed variant 362000
Processed variant 363000
Processed variant 364000
Processed variant 365000
Processed variant 366000
Processed variant 367000
Processed variant 368000
Processed variant 369000
Processed variant 370000
Processed variant 371000
Processed variant 372000
Processed variant 373000
Processed variant 374000
Processed variant 375000
Processed variant 376000
Processed variant 377000
Processed variant 378000
Processed variant 379000
Processed variant 380000
Processed variant 381000
Processed variant 382000
Processed variant 383000
Processed variant 384000
Processed variant 385000
Processed variant 386000
Processed variant 387000
Processed variant 388000
Processed variant 389000
Processed variant 390000
Processed variant 391000
Processed variant 392000
Processed variant 393000
Processed variant 394000
Processed variant 395000
Processed variant 396000
Processed variant 397000
Processed variant 398000
Processed variant 399000
Processed variant 400000
Processed variant 401000
Processed variant 402000
Processed variant 403000
Processed variant 404000
Processed variant 405000
Processed variant 406000
Processed variant 407000
Processed variant 408000
Processed variant 409000
Processed variant 410000
Processed variant 411000
Processed variant 412000
Processed variant 413000
Processed variant 414000
Processed variant 415000
Processed variant 416000
Processed variant 417000
Processed variant 418000
Processed variant 419000
Processed variant 420000
Processed variant 421000
Processed variant 422000
Processed variant 423000
Processed variant 424000
Processed variant 425000
Processed variant 426000
Processed variant 427000
Processed variant 428000
Processed variant 429000
Processed variant 430000
Processed variant 431000
Processed variant 432000
Processed variant 433000
Processed variant 434000
Processed variant 435000
Processed variant 436000
Processed variant 437000
Processed variant 438000
Processed variant 439000
Processed variant 440000
Processed variant 441000
Processed variant 442000
Processed variant 443000
Processed variant 444000
Processed variant 445000
Processed variant 446000
Processed variant 447000
Processed variant 448000
Processed variant 449000
Processed variant 450000
Processed variant 451000
Processed variant 452000
Processed variant 453000
Processed variant 454000
Processed variant 455000
Processed variant 456000
Processed variant 457000
Processed variant 458000
Processed variant 459000
Processed variant 460000
Processed variant 461000
Processed variant 462000
Processed variant 463000
Processed variant 464000
Processed variant 465000
Processed variant 466000
Processed variant 467000
Processed variant 468000
Processed variant 469000
Processed variant 470000
Processed variant 471000
Processed variant 472000
Processed variant 473000
Processed variant 474000
Processed variant 475000
Processed variant 476000
Processed variant 477000
Processed variant 478000
Processed variant 479000
Processed variant 480000
Processed variant 481000
Processed variant 482000
Processed variant 483000
Processed variant 484000
Processed variant 485000
Processed variant 486000
Processed variant 487000
Processed variant 488000
Processed variant 489000
Processed variant 490000
Processed variant 491000
Processed variant 492000
Processed variant 493000
Processed variant 494000
Processed variant 495000
Processed variant 496000
Processed variant 497000
Processed variant 498000
Processed variant 499000
Processed variant 500000
Processed variant: 500000
## All variants processed</code></pre>
<div class="sourceCode" id="cb137"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#now benchmark without having to read in the file itself</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/pkg/microbenchmark/man/microbenchmark.html" class="external-link">microbenchmark</a></span><span class="op">(</span></span>
<span>  <span class="co">#benchmark 500K while having to read in the vcf file</span></span>
<span>  <span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">x</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>,</span>
<span>  <span class="co">#set number of reps and units as seconds</span></span>
<span>  times <span class="op">=</span> <span class="fl">3</span>, unit <span class="op">=</span> <span class="st">"s"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">## 1.32% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.32% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 29.3% of genotypes fall below a read depth of 5 and were converted to NA</span></span></code></pre>
<pre><code><span><span class="co">## 1.32% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span></code></pre>
<div class="sourceCode" id="cb143"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#convert each to a dataframe</span></span>
<span><span class="va">y</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">y</span><span class="op">)</span></span>
<span><span class="va">z</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">z</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add these results into the full dataframe</span></span>
<span><span class="va">sum.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">sum.df</span>,<span class="va">y</span>,<span class="va">z</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="now-add-informative-columns-to-the-dataframe">Now add informative columns to the dataframe<a class="anchor" aria-label="anchor" href="#now-add-informative-columns-to-the-dataframe"></a>
</h4>
<div class="sourceCode" id="cb144"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sum.df</span><span class="op">$</span><span class="va">approach</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNPfiltR+vcfR"</span>,<span class="st">"SNPfiltR"</span><span class="op">)</span>, times <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sum.df</span><span class="op">$</span><span class="va">SNPs</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10000</span>,<span class="fl">10000</span>,<span class="fl">20000</span>,<span class="fl">20000</span>,<span class="fl">50000</span>,<span class="fl">50000</span>,<span class="fl">100000</span>,<span class="fl">100000</span>,</span>
<span>               <span class="fl">200000</span>,<span class="fl">200000</span>,<span class="fl">300000</span>,<span class="fl">300000</span>,<span class="fl">400000</span>,<span class="fl">400000</span>,<span class="fl">500000</span>,<span class="fl">500000</span><span class="op">)</span></span>
<span></span>
<span><span class="va">SNPfiltR.times</span><span class="op">&lt;-</span><span class="va">sum.df</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">9</span>,<span class="fl">10</span>,<span class="fl">4</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="now-time-vcftools-performing-the-same-filtering-in-bash-three-times-for-the-10k-snp-dataset">Now time VCFtools performing the same filtering, in bash, three
times for the 10K SNP dataset<a class="anchor" aria-label="anchor" href="#now-time-vcftools-performing-the-same-filtering-in-bash-three-times-for-the-10k-snp-dataset"></a>
</h4>
</div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-10k">VCFtools benchmark 10K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-10k"></a>
</h2>
<div class="sourceCode" id="cb145"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;</span> time.txt</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-6"><a href="#cb145-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.10K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-20k">VCFtools benchmark 20K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-20k"></a>
</h2>
<div class="sourceCode" id="cb146"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb146-1"><a href="#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb146-2"><a href="#cb146-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb146-3"><a href="#cb146-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-4"><a href="#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb146-5"><a href="#cb146-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-6"><a href="#cb146-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.20K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-50k">VCFtools benchmark 50K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-50k"></a>
</h2>
<div class="sourceCode" id="cb147"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb147-2"><a href="#cb147-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb147-3"><a href="#cb147-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-4"><a href="#cb147-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb147-5"><a href="#cb147-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-6"><a href="#cb147-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.50K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-100k">VCFtools benchmark 100K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-100k"></a>
</h2>
<div class="sourceCode" id="cb148"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.100K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.100K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.100K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-200k">VCFtools benchmark 200K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-200k"></a>
</h2>
<div class="sourceCode" id="cb149"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.200K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.200K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb149-5"><a href="#cb149-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-6"><a href="#cb149-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.200K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-300k">VCFtools benchmark 300K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-300k"></a>
</h2>
<div class="sourceCode" id="cb150"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb150-1"><a href="#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb150-2"><a href="#cb150-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.300K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb150-3"><a href="#cb150-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-4"><a href="#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.300K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb150-5"><a href="#cb150-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-6"><a href="#cb150-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.300K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-400k">VCFtools benchmark 400K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-400k"></a>
</h2>
<div class="sourceCode" id="cb151"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.400K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.400K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.400K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="vcftools-benchmark-500k">VCFtools benchmark 500K<a class="anchor" aria-label="anchor" href="#vcftools-benchmark-500k"></a>
</h2>
<div class="sourceCode" id="cb152"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.500K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb152-3"><a href="#cb152-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-4"><a href="#cb152-4" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.500K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span>
<span id="cb152-5"><a href="#cb152-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-6"><a href="#cb152-6" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span> <span class="bu">time</span> /Users/devder/Downloads/vcftools/src/cpp/vcftools <span class="at">--gzvcf</span> benchmark.500K.vcf.gz <span class="at">--minGQ</span> 30 <span class="at">--minDP</span> 5 <span class="at">--recode</span> <span class="at">--recode-INFO-all</span> <span class="kw">;</span> <span class="kw">}</span> <span class="dv">2</span><span class="op">&gt;&amp;</span><span class="dv">1</span> <span class="kw">|</span>  <span class="fu">cat</span> <span class="op">&gt;&gt;</span> time.txt</span></code></pre></div>
<div class="section level4">
<h4 id="clean-output-file-with-sed-one-liner-to-keep-only-the-lines-beginning-with-real-which-contain-the-time-info-for-each-replicate-run">clean output file with sed one-liner to keep only the lines
beginning with “real” (which contain the time info for each replicate
run)<a class="anchor" aria-label="anchor" href="#clean-output-file-with-sed-one-liner-to-keep-only-the-lines-beginning-with-real-which-contain-the-time-info-for-each-replicate-run"></a>
</h4>
<div class="sourceCode" id="cb153"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /Users/devder/Desktop/benchmarking.vcfs</span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a><span class="co">#use sed one-liner to get only the times of each run in a single file</span></span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-n</span> <span class="at">-e</span> <span class="st">'/^real/ p'</span> time.txt <span class="op">&gt;</span> cleantimes.txt</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="read-cleaned-file-into-r-and-manipulate-into-a-cohesive-dataframe-with-our-microbenchmark-output-dataframe">read cleaned file into R and manipulate into a cohesive dataframe
with our microbenchmark output dataframe<a class="anchor" aria-label="anchor" href="#read-cleaned-file-into-r-and-manipulate-into-a-cohesive-dataframe-with-our-microbenchmark-output-dataframe"></a>
</h4>
<div class="sourceCode" id="cb154"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in VCFtools times</span></span>
<span><span class="va">vcftools.times</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"~/Desktop/benchmarking.vcfs/cleantimes.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#make column 1 informative about run conditions</span></span>
<span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V1</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">10000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">20000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">50000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">100000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">200000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">300000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,</span>
<span>                     <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">400000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">500000</span>, times<span class="op">=</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#format column 2</span></span>
<span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"m"</span>, <span class="st">":"</span>, <span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span><span class="op">)</span></span>
<span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"s"</span>, <span class="st">""</span>, <span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span><span class="op">)</span></span>
<span><span class="co">#vcftools.times$V2&lt;-as.numeric(as.character(vcftools.times$V2))</span></span>
<span><span class="co">#convert minutes:seconds to just seconds as.numeric for plotting</span></span>
<span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span><span class="op">&lt;-</span><span class="fu">lubridate</span><span class="fu">::</span><span class="fu"><a href="https://lubridate.tidyverse.org/reference/period_to_seconds.html" class="external-link">period_to_seconds</a></span><span class="op">(</span><span class="fu">lubridate</span><span class="fu">::</span><span class="fu"><a href="https://lubridate.tidyverse.org/reference/hms.html" class="external-link">ms</a></span><span class="op">(</span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#get the mean of the runtimes across all three replicates, for each vcf file</span></span>
<span><span class="va">v</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/stats/aggregate.html" class="external-link">aggregate</a></span><span class="op">(</span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V2</span>, <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">vcftools.times</span><span class="op">$</span><span class="va">V1</span><span class="op">)</span>, <span class="va">mean</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#add column tagging all of these times as coming from vcftools</span></span>
<span><span class="va">v</span><span class="op">$</span><span class="va">approach</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"VCFtools"</span>, times<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#give informative column names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">v</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNPs"</span>,<span class="st">"mean"</span><span class="op">)</span></span>
<span><span class="co">#reorder to match column order from microbenchmark</span></span>
<span><span class="va">v</span><span class="op">&lt;-</span><span class="va">v</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>,<span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">#rbind dataframe with the microbenchmark dataframe</span></span>
<span><span class="va">benchmark.df</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">SNPfiltR.times</span>,<span class="va">v</span><span class="op">)</span></span>
<span><span class="co">#make approach a factor for plotting</span></span>
<span><span class="va">benchmark.df</span><span class="op">$</span><span class="va">approach</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">benchmark.df</span><span class="op">$</span><span class="va">approach</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#final product:</span></span>
<span><span class="va">benchmark.df</span></span></code></pre></div>
<pre><code><span><span class="co">##         approach  SNPs        mean</span></span>
<span><span class="co">## 1  SNPfiltR+vcfR 1e+04   1.3929194</span></span>
<span><span class="co">## 2       SNPfiltR 1e+04   0.4061834</span></span>
<span><span class="co">## 3  SNPfiltR+vcfR 2e+04   2.7607792</span></span>
<span><span class="co">## 4       SNPfiltR 2e+04   0.9804999</span></span>
<span><span class="co">## 5  SNPfiltR+vcfR 5e+04   6.6884265</span></span>
<span><span class="co">## 6       SNPfiltR 5e+04   2.6912465</span></span>
<span><span class="co">## 7  SNPfiltR+vcfR 1e+05  15.1215112</span></span>
<span><span class="co">## 8       SNPfiltR 1e+05   5.5519810</span></span>
<span><span class="co">## 9  SNPfiltR+vcfR 2e+05  34.3970906</span></span>
<span><span class="co">## 10      SNPfiltR 2e+05  13.4741584</span></span>
<span><span class="co">## 11 SNPfiltR+vcfR 3e+05  48.4870002</span></span>
<span><span class="co">## 12      SNPfiltR 3e+05  18.2342997</span></span>
<span><span class="co">## 13 SNPfiltR+vcfR 4e+05  69.9663987</span></span>
<span><span class="co">## 14      SNPfiltR 4e+05  28.6984442</span></span>
<span><span class="co">## 15 SNPfiltR+vcfR 5e+05 101.2610619</span></span>
<span><span class="co">## 16      SNPfiltR 5e+05  37.9307606</span></span>
<span><span class="co">## 17      VCFtools 1e+04   1.8620000</span></span>
<span><span class="co">## 18      VCFtools 2e+04   2.3426667</span></span>
<span><span class="co">## 19      VCFtools 5e+04   5.1456667</span></span>
<span><span class="co">## 20      VCFtools 1e+05  10.5826667</span></span>
<span><span class="co">## 21      VCFtools 2e+05  22.0676667</span></span>
<span><span class="co">## 22      VCFtools 3e+05  41.0583333</span></span>
<span><span class="co">## 23      VCFtools 4e+05  50.1796667</span></span>
<span><span class="co">## 24      VCFtools 5e+05  53.2986667</span></span></code></pre>
<div class="sourceCode" id="cb156"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">benchmark.df</span>, file<span class="op">=</span><span class="st">"~/Desktop/benchmarking.vcfs/benchmark.info.csv"</span>, quote <span class="op">=</span> <span class="cn">F</span>, row.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="compare-runtimes">Compare runtimes<a class="anchor" aria-label="anchor" href="#compare-runtimes"></a>
</h2>
<div class="sourceCode" id="cb157"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#visualize the comparative runtimes across SNPs and between approaches</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">benchmark.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">SNPs</span>, y <span class="op">=</span> <span class="va">mean</span>, color <span class="op">=</span> <span class="va">approach</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">approach</span><span class="op">)</span>,size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">approach</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"mean runtime (seconds)"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="performance-benchmarking_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
<div class="sourceCode" id="cb158"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">runtime.plot</span><span class="op">&lt;-</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">benchmark.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">SNPs</span>, y <span class="op">=</span> <span class="va">mean</span>, color <span class="op">=</span> <span class="va">approach</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">approach</span><span class="op">)</span>,size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">approach</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"mean runtime (seconds)"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html" class="external-link">ggsave</a></span><span class="op">(</span><span class="va">runtime.plot</span>,</span>
<span>      filename <span class="op">=</span> <span class="st">"~/Desktop/SNPfiltR.mol.ecol.resour.submission/comp.runtimes.pdf"</span>,</span>
<span>      height <span class="op">=</span> <span class="fl">3</span>, width <span class="op">=</span> <span class="fl">4</span>, units <span class="op">=</span> <span class="st">"in"</span><span class="op">)</span></span>
<span>  </span>
<span></span>
<span><span class="co">#try it with a log10 scaled x axis</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">benchmark.df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">SNPs</span>, y <span class="op">=</span> <span class="va">mean</span>, color <span class="op">=</span> <span class="va">approach</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">approach</span><span class="op">)</span>,size<span class="op">=</span><span class="fl">3</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html" class="external-link">geom_line</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>group <span class="op">=</span> <span class="va">approach</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_classic</a></span><span class="op">(</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"mean runtime (seconds)"</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="performance-benchmarking_files/figure-html/unnamed-chunk-21-2.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
