<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>reproducible-vignette • SNPfiltR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="reproducible-vignette">
<meta property="og:description" content="SNPfiltR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/accuracy-benchmarking.html">accuracy-benchmarking</a>
    </li>
    <li>
      <a href="../articles/performance-benchmarking.html">Performance benchmarking</a>
    </li>
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-RADseq-vignette.html">scrub-jay-RADseq-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-UCE-vignette.html">scrub-jay-UCE-vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="reproducible-vignette_files/header-attrs-2.10/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>reproducible-vignette</h1>
            
      
      
      <div class="hidden name"><code>reproducible-vignette.Rmd</code></div>

    </div>

    
    
<p>This is a small, quick-running example showing how to use the SNPfiltR package to filter a next-generation sequencing (NGS) single nucleotide polymorphism (SNP) dataset. This vignette uses the small example dataset distributed with the package (20 samples, and 500 SNPs). The entire vignette can be rendered in &lt;60s, allowing rapid validation of the filtering pipeline shown here. Overall, this pipeline shows the power of SNPfiltR to facilitate the visualization, exploration, and interactive filtering of any SNP dataset. We start by reading in our vcf file as a vcfR object. Having the vcfR object stored in local memory allows for the rapid visualization and imposition of various filtering criteria, allowing users to make informed decisions and interactively explore parameter space. This vignette follows best-practices by implementing quality aware filters (i.e., filters based on genotype quality) first, and quality blind filters (e.g., distance-based thinning) last, in order to preserve as many high-quality genotypes as possible for downstream analyses.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">SNPfiltR</span><span class="op">)</span>
<span class="co">#&gt; This is SNPfiltR v.1.0.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Detailed usage information is available at: devonderaad.github.io/SNPfiltR/ </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; If you use SNPfiltR in your published work, please cite the following papers: </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; DeRaad, D.A. (2022), SNPfiltR: an R package for interactive and reproducible SNP filtering. Molecular Ecology Resources, 00, 1–15. http://doi.org/10.1111/1755-0998.13618 </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Knaus, Brian J., and Niklaus J. Grunwald. 2017. VCFR: a package to manipulate and visualize variant call format data in R. Molecular Ecology Resources, 17.1:44-53. http://dx.doi.org/10.1111/1755-0998.12549</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR">vcfR</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;    *****       ***   vcfR   ***       *****</span>
<span class="co">#&gt;    This is vcfR 1.12.0 </span>
<span class="co">#&gt;      browseVignettes('vcfR') # Documentation</span>
<span class="co">#&gt;      citation('vcfR') # Citation</span>
<span class="co">#&gt;    *****       *****      *****       *****</span></code></pre></div>
<div id="optional-step-0" class="section level1">
<h1 class="hasAnchor">
<a href="#optional-step-0" class="anchor"></a>Optional Step 0:</h1>
<p>Do quality control per sample before performing SNP calling. I have written an <a href="https://github.com/DevonDeRaad/RADstackshelpR/blob/master/inst/extdata/fastqcr.Rmd">RMarkdown script</a> that uses the R package <a href="https://github.com/kassambara/fastqcr">fastqcr</a> to generate a report visualizing the quality and quantity of sequencing for each sample, and recommending a subset of samples to be immediately dropped before parameter optimization (specifically useful for RADseq data). The only modification necessary for this script is the path to the folder containing the input .fastq.gz files and the path to your desired output folder. An example report generated using this script can be seen <a href="https://devonderaad.github.io/RADstackshelpR/articles/quality.control.vignette.html">here</a>. Because the fastq.gz files for your experiment may be large and handled remotely, an example bash script for executing this RMarkdown file as a job on a high performance computing cluster is available <a href="https://github.com/DevonDeRaad/RADstackshelpR/blob/master/inst/extdata/RMarkdown.qc.submit.script.sh">here</a>.</p>
</div>
<div id="step-1-read-in-vcf-file-as-vcfr-object" class="section level1">
<h1 class="hasAnchor">
<a href="#step-1-read-in-vcf-file-as-vcfr-object" class="anchor"></a>Step 1: read in vcf file as ‘vcfR’ object</h1>
<p>Because we are using the example dataset distributed with the package, we just load in the vcf and popmap using the data() function.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load the example vcfR object </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">vcfR.example</span><span class="op">)</span>

<span class="co">### check the metadata present in your vcf</span>
<span class="va">vcfR.example</span>
<span class="co">#&gt; ***** Object of Class vcfR *****</span>
<span class="co">#&gt; 20 samples</span>
<span class="co">#&gt; 1 CHROMs</span>
<span class="co">#&gt; 500 variants</span>
<span class="co">#&gt; Object size: 0.7 Mb</span>
<span class="co">#&gt; 38.07 percent missing data</span>
<span class="co">#&gt; *****        *****         *****</span>

<span class="va">vcfR.example</span><span class="op">@</span><span class="va">fix</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>
<span class="co">#&gt;       CHROM        POS        ID           REF ALT QUAL FILTER</span>
<span class="co">#&gt;  [1,] "Pseudochr1" "25801497" "16579:67:+" "A" "G" NA   "PASS"</span>
<span class="co">#&gt;  [2,] "Pseudochr1" "25457396" "16292:5:-"  "A" "G" NA   "PASS"</span>
<span class="co">#&gt;  [3,] "Pseudochr1" "20817192" "12868:12:-" "C" "T" NA   "PASS"</span>
<span class="co">#&gt;  [4,] "Pseudochr1" "29813636" "19560:94:+" "T" "C" NA   "PASS"</span>
<span class="co">#&gt;  [5,] "Pseudochr1" "28229092" "18530:62:-" "A" "G" NA   "PASS"</span>
<span class="co">#&gt;  [6,] "Pseudochr1" "9354478"  "4669:39:+"  "C" "T" NA   "PASS"</span>
<span class="co">#&gt;  [7,] "Pseudochr1" "17764465" "10787:26:+" "C" "T" NA   "PASS"</span>
<span class="co">#&gt;  [8,] "Pseudochr1" "5349729"  "1884:88:-"  "A" "T" NA   "PASS"</span>
<span class="co">#&gt;  [9,] "Pseudochr1" "17438085" "10478:65:-" "T" "G" NA   "PASS"</span>
<span class="co">#&gt; [10,] "Pseudochr1" "30854925" "20439:79:+" "G" "T" NA   "PASS"</span>
<span class="co">#&gt;       INFO             </span>
<span class="co">#&gt;  [1,] "NS=93;AF=0.059" </span>
<span class="co">#&gt;  [2,] "NS=54;AF=0.056" </span>
<span class="co">#&gt;  [3,] "NS=109;AF=0.037"</span>
<span class="co">#&gt;  [4,] "NS=63;AF=0.016" </span>
<span class="co">#&gt;  [5,] "NS=97;AF=0.433" </span>
<span class="co">#&gt;  [6,] "NS=17;AF=0.059" </span>
<span class="co">#&gt;  [7,] "NS=103;AF=0.330"</span>
<span class="co">#&gt;  [8,] "NS=9;AF=0.111"  </span>
<span class="co">#&gt;  [9,] "NS=16;AF=0.469" </span>
<span class="co">#&gt; [10,] "NS=102;AF=0.010"</span>

<span class="va">vcfR.example</span><span class="op">@</span><span class="va">gt</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>
<span class="co">#&gt;       FORMAT           A_californica_333849                     </span>
<span class="co">#&gt;  [1,] "GT:DP:AD:GQ:GL" "0/0:44:44,0:40:-0.00,-14.55,-152.08"    </span>
<span class="co">#&gt;  [2,] "GT:DP:AD:GQ:GL" NA                                       </span>
<span class="co">#&gt;  [3,] "GT:DP:AD:GQ:GL" "0/0:347:347,0:40:-0.00,-105.46,-1205.89"</span>
<span class="co">#&gt;  [4,] "GT:DP:AD:GQ:GL" "1/1:3:0,3:16:-9.57,-1.17,-0.03"         </span>
<span class="co">#&gt;  [5,] "GT:DP:AD:GQ:GL" "0/1:67:33,34:40:-78.52,0.00,-75.77"     </span>
<span class="co">#&gt;  [6,] "GT:DP:AD:GQ:GL" NA                                       </span>
<span class="co">#&gt;  [7,] "GT:DP:AD:GQ:GL" "0/0:331:331,0:40:0.00,-99.80,-1063.41"  </span>
<span class="co">#&gt;  [8,] "GT:DP:AD:GQ:GL" NA                                       </span>
<span class="co">#&gt;  [9,] "GT:DP:AD:GQ:GL" NA                                       </span>
<span class="co">#&gt; [10,] "GT:DP:AD:GQ:GL" "0/0:35:35,0:40:-0.00,-12.08,-112.81"</span>

<span class="co">#Load the example popmap file. It is a standard two column popmap, where the first column must be named 'id' and contain individual sample identifiers matching the sample identifiers in the vcf file, and the second column must be named 'pop', and contain a population assignment for each sample.</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">popmap</span><span class="op">)</span>
<span class="va">popmap</span>
<span class="co">#&gt;                       id          pop</span>
<span class="co">#&gt; 1   A_californica_333849  californica</span>
<span class="co">#&gt; 2   A_californica_333854  californica</span>
<span class="co">#&gt; 3   A_californica_333855  californica</span>
<span class="co">#&gt; 4   A_californica_333857  californica</span>
<span class="co">#&gt; 5   A_californica_333860  californica</span>
<span class="co">#&gt; 6     A_insularis_334031    insularis</span>
<span class="co">#&gt; 7     A_insularis_334032    insularis</span>
<span class="co">#&gt; 8     A_insularis_334033    insularis</span>
<span class="co">#&gt; 9     A_insularis_334034    insularis</span>
<span class="co">#&gt; 10    A_insularis_334037    insularis</span>
<span class="co">#&gt; 11  A_woodhouseii_334133  woodhouseii</span>
<span class="co">#&gt; 12  A_woodhouseii_334134  woodhouseii</span>
<span class="co">#&gt; 13  A_woodhouseii_334142  woodhouseii</span>
<span class="co">#&gt; 14  A_woodhouseii_334148  woodhouseii</span>
<span class="co">#&gt; 15  A_woodhouseii_334153  woodhouseii</span>
<span class="co">#&gt; 16 A_coerulescens_396256 coerulescens</span>
<span class="co">#&gt; 17 A_coerulescens_396259 coerulescens</span>
<span class="co">#&gt; 18 A_coerulescens_396262 coerulescens</span>
<span class="co">#&gt; 19 A_coerulescens_396263 coerulescens</span>
<span class="co">#&gt; 20 A_coerulescens_396264 coerulescens</span></code></pre></div>
</div>
<div id="step-2-quality-filtering" class="section level1">
<h1 class="hasAnchor">
<a href="#step-2-quality-filtering" class="anchor"></a>Step 2: quality filtering</h1>
<p>We now implement quality filters that don’t involve missing data. This is because removing low data samples will alter percentage/quantile based missing data cutoffs, so we wait to implement those until after deciding on our final set of samples for downstream analysis</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#generate exploratory visualizations of depth and genotype quality for all called genotypes</span>
<span class="co">#hard_filter(vcfR=vcfR.example)</span>

<span class="co">#hard filter to minimum depth of 5, and minimum genotype quality of 30</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR.example</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span>
<span class="co">#&gt; 28.81% of genotypes fall below a read depth of 5 and were converted to NA</span>
<span class="co">#&gt; 3.81% of genotypes fall below a genotype quality of 30 and were converted to NA</span></code></pre></div>
<p>Jon Puritz has an excellent filtering tutorial that is focused specifically on <a href="https://www.ddocent.com/filtering/">filtering RADseq data</a>.From Puritz SNP filtering tutorial “Allele balance: a number between 0 and 1 representing the ratio of reads showing the reference allele to all reads, considering only reads from individuals called as heterozygous, we expect that the allele balance in our data (for real loci) should be close to 0.5”. Here we will implement an allele balance filter converting called heterozygous genotypes outside of the .25-.75 range to NA.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#execute allele balance filter</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/filter_allele_balance.html">filter_allele_balance</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span>
<span class="co">#&gt; 7.54% of het genotypes (0.71% of all genotypes) fall outside of 0.25 - 0.75 allele balance ratio and were converted to NA</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-4-1.png" width="384"></p>
<p>Here we will implement a max depth filter, as super high depth loci are likely multiple loci stuck together into a single paralogous locus, which we want to remove before making downstream inferences.</p>
<div id="note" class="section level3">
<h3 class="hasAnchor">
<a href="#note" class="anchor"></a>Note:</h3>
<p>This filter is applied ‘per SNP’ rather than ‘per genotype’ otherwise we would simply be removing most of the genotypes from our deepest sequenced samples (because sequencing depth is so variable between samples). By filtering per SNP, we remove the SNPs with outlier depth values, which are most likely to be spuriously mapped/built paralagous loci.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize and pick appropriate max depth cutoff</span>
<span class="co">#max_depth(vcfR)</span>
<span class="co">#not running here to save space on visualizations</span>

<span class="co">#filter vcf by the max depth cutoff you chose</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/max_depth.html">max_depth</a></span><span class="op">(</span><span class="va">vcfR</span>, maxdepth <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="co">#&gt; maxdepth cutoff is specified, filtered vcfR object will be returned</span>
<span class="co">#&gt; 17.6% of SNPs were above a mean depth of 100 and were removed from the vcf</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-5-1.png" width="384"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#check vcfR to see how many SNPs we have left</span>
<span class="va">vcfR</span>
<span class="co">#&gt; ***** Object of Class vcfR *****</span>
<span class="co">#&gt; 20 samples</span>
<span class="co">#&gt; 2 CHROMs</span>
<span class="co">#&gt; 412 variants</span>
<span class="co">#&gt; Object size: 0.4 Mb</span>
<span class="co">#&gt; 67.65 percent missing data</span>
<span class="co">#&gt; *****        *****         *****</span></code></pre></div>
</div>
</div>
<div id="step-3-set-missing-data-per-sample-cutoff" class="section level1">
<h1 class="hasAnchor">
<a href="#step-3-set-missing-data-per-sample-cutoff" class="anchor"></a>Step 3: set missing data per sample cutoff</h1>
<div id="set-arbitrary-cutoff-for-missing-data-allowed-per-sample-" class="section level3">
<h3 class="hasAnchor">
<a href="#set-arbitrary-cutoff-for-missing-data-allowed-per-sample-" class="anchor"></a>Set arbitrary cutoff for missing data allowed per sample.</h3>
<p>Determining which samples and SNPs to retain is always project specific, and is contingent on sampling, biology of the focal taxa, sequencing idiosyncrasies, etc. SNPfiltR contains functions designed to simply and easily generate exploratory visualizations that will allow you to make informed decisions about which samples and SNPs are of sufficient quality to retain for downstream analyses, but there is never a single correct option for these cutoffs. In my experience, the best thing to do is to look at your data, look at what effects some reasonable cutoffs would have on your data, and pick one that works for you. Then as you continue to analyze your data, make sure that your arbitrary filtering decisions are not driving the patterns you see, and iteratively update your filtering approach if you are concerned about the effects previous filtering choices are having on downstream results.</p>
<p>We are going to start by visualizing missing data per sample. After checking out the visualizations, we can make decision on which samples look like they will not be salvageable for downstream analysis, and remove them either by setting a data completeness cutoff in the function missing_by_sample(), or by name using base R syntax which works with vcfR objects which treat SNPs as rows and samples as columns (e.g., vcfR &lt;- vcfR[,colnames(<a href="mailto:vcfR@gt" class="email">vcfR@gt</a>) != “A_woodhouseii_24711”]).</p>
</div>
<div id="note-1" class="section level3">
<h3 class="hasAnchor">
<a href="#note-1" class="anchor"></a>Note:</h3>
<p>If all of your samples are reasonably complete then you don’t need to drop any samples! Variable missing data by sample seems to be mostly an issue with RAD approaches for whatever reason.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#run function to visualize samples and return informative data.frame object</span>
<span class="va">miss</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_sample.html">missing_by_sample</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span><span class="op">)</span>
<span class="co">#&gt; No popmap provided</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-6-1.png" width="384"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-6-2.png" width="384"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#run function to drop samples above the threshold we want from the vcf</span>
<span class="co">#here I am setting a relatively lax cutoff</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_sample.html">missing_by_sample</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, cutoff <span class="op">=</span> <span class="fl">.9</span><span class="op">)</span>
<span class="co">#&gt; 3 samples are above a 0.9 missing data cutoff, and were removed from VCF</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-6-3.png" width="384"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#remove invariant sites generated by sample trimming and genotype filtering</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span><span class="va">vcfR</span>, min.mac <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="co">#&gt; 51.94% of SNPs fell below a minor allele count of 1 and were removed from the VCF</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-6-4.png" width="384"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">#update popmap by removing samples that have been filtered out</span>
<span class="va">popmap</span><span class="op">&lt;-</span><span class="va">popmap</span><span class="op">[</span><span class="va">popmap</span><span class="op">$</span><span class="va">id</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,<span class="op">]</span></code></pre></div>
</div>
</div>
<div id="step-4-set-missing-data-per-snp-cutoff" class="section level1">
<h1 class="hasAnchor">
<a href="#step-4-set-missing-data-per-snp-cutoff" class="anchor"></a>Step 4: set missing data per SNP cutoff</h1>
<div id="set-arbitrary-cutoff-for-missing-data-allowed-per-snp-" class="section level3">
<h3 class="hasAnchor">
<a href="#set-arbitrary-cutoff-for-missing-data-allowed-per-snp-" class="anchor"></a>Set arbitrary cutoff for missing data allowed per SNP.</h3>
<p>We can visualize the effect that typical missing data cutoffs will have on both the number of SNPs retained and the total missing data in our entire dataset.We want to choose a cutoff that minimizes the overall missing data in the dataset, while maximizing the total number of loci retained.</p>
</div>
<div id="note-2" class="section level3">
<h3 class="hasAnchor">
<a href="#note-2" class="anchor"></a>Note:</h3>
<p>This filter interacts with the above filter, where we dropped low data samples. A good rule of thumb is that individual samples shouldn’t be above 50% missing data after applying a per-SNP missing data cutoff. So if we are retaining specific low data samples out of necessity or project design, we may have to set a more stringent per-SNP missing data cutoff, at the expense of the total number of SNPs retained for downstream analyses. We can again use the assess_missing_data_pca() function to determine whether all retained samples contain enough data at our chosen cutoff in order to be assigned accurately to their species group.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize missing data by SNP and the effect of various cutoffs on the missingness of each sample</span>
<span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span>
<span class="co">#&gt; cutoff is not specified, exploratory visualizations will be generated</span>
<span class="co">#&gt; Picking joint bandwidth of 0.128</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-7-1.png" width="384"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-7-2.png" width="384"></p>
<pre><code>#&gt;    filt missingness snps.retained
#&gt; 1  0.30  0.26907021           155
#&gt; 2  0.50  0.20036765           128
#&gt; 3  0.60  0.15631808           108
#&gt; 4  0.65  0.13844326            99
#&gt; 5  0.70  0.13844326            99
#&gt; 6  0.75  0.12095175            89
#&gt; 7  0.80  0.08991597            70
#&gt; 8  0.85  0.06427015            54
#&gt; 9  0.90  0.02466793            31
#&gt; 10 0.95  0.00000000            18
#&gt; 11 1.00  0.00000000            18</code></pre>
<p>We can see that there are still some outlier samples with a lot of missing data even at high missing data per SNP thresholds, which is concerning. We will want to check whether this excess missing data in some samples is affecting overall clustering patterns using the functions assess_missing_data_pca() and assess_missing_data_tsne().</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#assess missing data effects on clustering</span>
<span class="fu"><a href="../reference/assess_missing_data_pca.html">assess_missing_data_pca</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">vcfR</span>, popmap <span class="op">=</span> <span class="va">popmap</span>, thresholds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.8</span><span class="op">)</span>, clustering <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span>
<span class="co">#&gt; 64.65% of SNPs fell below a completeness cutoff of 0.8 and were removed from the VCF</span>
<span class="co">#&gt; Loading required namespace: adegenet</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-8-1.png" width="700"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-8-2.png" width="700"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-8-3.png" width="700"></p>
<pre><code><a href="../reference/assess_missing_data_tsne.html">#&gt; [[1]]
#&gt;                              PC1        PC2          PC3         PC4
#&gt; A_californica_333849  -0.5969622 -0.5362690 -0.595508158  0.73945078
#&gt; A_californica_333854  -0.4693982 -0.9890245 -1.414869592 -1.37005171
#&gt; A_californica_333855   0.5581980 -0.1075819 -0.257991833  0.14855832
#&gt; A_californica_333857  -1.0182608 -0.5641853 -0.417897234 -0.28592615
#&gt; A_californica_333860  -0.5224670 -0.7392822 -0.823298194 -0.01861634
#&gt; A_insularis_334031     1.6356477  0.3136035  0.074824184 -0.03538247
#&gt; A_insularis_334032     1.6717696  0.3245417  0.079829224 -0.03590089
#&gt; A_insularis_334033     1.6695981  0.3219187  0.075663814 -0.03325062
#&gt; A_insularis_334034     1.6356477  0.3136035  0.074824184 -0.03538247
#&gt; A_insularis_334037     1.6356477  0.3136035  0.074824184 -0.03538247
#&gt; A_woodhouseii_334133  -0.5830285 -0.1164268 -0.007498313  0.86818251
#&gt; A_woodhouseii_334134  -0.4918528 -0.6028390  0.150783442  0.31303844
#&gt; A_woodhouseii_334142  -0.8005397 -1.0392687  0.531437285  1.20318761
#&gt; A_woodhouseii_334153  -0.6375918 -1.3648416  2.046665190 -1.05594446
#&gt; A_coerulescens_396256 -1.1495480  1.3483650  0.060491990  0.14644120
#&gt; A_coerulescens_396263 -0.8159388  0.6141574  0.121397651 -0.06714767
#&gt; A_coerulescens_396264 -1.7209210  2.5099257  0.226322176 -0.44587363
#&gt;                               PC5          PC6          pop    missing
#&gt; A_californica_333849   1.34940006 -0.217111195  californica 0.01428571
#&gt; A_californica_333854  -0.31732671 -0.252236192  californica 0.08571429
#&gt; A_californica_333855   0.79002290 -0.411529451  californica 0.07142857
#&gt; A_californica_333857  -0.27279109 -0.675313989  californica 0.00000000
#&gt; A_californica_333860  -0.29414891  1.230546314  californica 0.01428571
#&gt; A_insularis_334031    -0.11699427  0.010904247    insularis 0.00000000
#&gt; A_insularis_334032    -0.12438335  0.006458987    insularis 0.00000000
#&gt; A_insularis_334033    -0.11649650 -0.001424441    insularis 0.01428571
#&gt; A_insularis_334034    -0.11699427  0.010904247    insularis 0.00000000
#&gt; A_insularis_334037    -0.11699427  0.010904247    insularis 0.00000000
#&gt; A_woodhouseii_334133   0.22400118  0.462404897  woodhouseii 0.01428571
#&gt; A_woodhouseii_334134  -0.05571374  0.543962142  woodhouseii 0.48571429
#&gt; A_woodhouseii_334142  -1.18389174 -0.521504387  woodhouseii 0.04285714
#&gt; A_woodhouseii_334153   0.44668940  0.174951057  woodhouseii 0.04285714
#&gt; A_coerulescens_396256  0.01905756  0.014972030 coerulescens 0.21428571
#&gt; A_coerulescens_396263  0.03051154 -0.599271664 coerulescens 0.51428571
#&gt; A_coerulescens_396264 -0.14394778  0.212383154 coerulescens 0.01428571
assess_missing_data_tsne(vcfR = vcfR, popmap = popmap, thresholds = c(.8), clustering = FALSE)
#&gt; cutoff is specified, filtered vcfR object will be returned
#&gt; 64.65% of SNPs fell below a completeness cutoff of 0.8 and were removed from the VCF</a></code></pre>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-8-4.png" width="700"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-8-5.png" width="700"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-8-6.png" width="700"></p>
<pre><code>#&gt; [[1]]
#&gt;            V1         V2          pop    missing
#&gt; 1  -27.699440   0.504800  californica 0.01428571
#&gt; 2  -73.726672  -7.530026  californica 0.08571429
#&gt; 3   -8.447948  -5.695262  californica 0.07142857
#&gt; 4  -50.710891 -17.334099  californica 0.00000000
#&gt; 5  -62.916094 -23.004145  californica 0.01428571
#&gt; 6   85.199605  72.888293    insularis 0.00000000
#&gt; 7   64.054586  57.926042    insularis 0.00000000
#&gt; 8   64.367387  72.037898    insularis 0.01428571
#&gt; 9   74.090496  85.953990    insularis 0.00000000
#&gt; 10  83.900058  57.102125    insularis 0.00000000
#&gt; 11 -25.286082 -20.828195  woodhouseii 0.01428571
#&gt; 12 -37.590537 -31.280806  woodhouseii 0.48571429
#&gt; 13 -56.718615 -46.367156  woodhouseii 0.04285714
#&gt; 14 -33.809945 -54.868297  woodhouseii 0.04285714
#&gt; 15   4.767038 -42.879079 coerulescens 0.21428571
#&gt; 16  -9.058097 -38.997170 coerulescens 0.51428571
#&gt; 17   9.585150 -57.628913 coerulescens 0.01428571</code></pre>
<p>We can see that at an 80% per SNP completeness cutoff, samples with an excess of missing data are leaking toward the center (specifically coerulescens samples), indicating that they can’t be reliably clustered due to excess missing data.</p>
<p>We can go back and target specific samples with too much missing data even at high filtering thresholds for removal using the following code:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#show me the samples with the most missing data at an 80% completeness threshold</span>
<span class="va">filt</span><span class="op">&lt;-</span><span class="va">miss</span><span class="op">[</span><span class="va">miss</span><span class="op">$</span><span class="va">filt</span> <span class="op">==</span> <span class="fl">.8</span>,<span class="op">]</span>
<span class="va">filt</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">filt</span><span class="op">$</span><span class="va">snps.retained</span><span class="op">)</span>,<span class="op">]</span>
<span class="co">#&gt;                    indiv filt snps.retained</span>
<span class="co">#&gt; 78 A_coerulescens_396262  0.8             0</span>
<span class="co">#&gt; 77 A_coerulescens_396259  0.8             6</span>
<span class="co">#&gt; 74  A_woodhouseii_334148  0.8            19</span>
<span class="co">#&gt; 79 A_coerulescens_396263  0.8            20</span>
<span class="co">#&gt; 72  A_woodhouseii_334134  0.8            26</span>
<span class="co">#&gt; 61  A_californica_333849  0.8            28</span>
<span class="co">#&gt; 62  A_californica_333854  0.8            29</span>
<span class="co">#&gt; 63  A_californica_333855  0.8            29</span>
<span class="co">#&gt; 64  A_californica_333857  0.8            29</span>
<span class="co">#&gt; 65  A_californica_333860  0.8            29</span>
<span class="co">#&gt; 66    A_insularis_334031  0.8            29</span>
<span class="co">#&gt; 67    A_insularis_334032  0.8            29</span>
<span class="co">#&gt; 68    A_insularis_334033  0.8            29</span>
<span class="co">#&gt; 69    A_insularis_334034  0.8            29</span>
<span class="co">#&gt; 70    A_insularis_334037  0.8            29</span>
<span class="co">#&gt; 71  A_woodhouseii_334133  0.8            29</span>
<span class="co">#&gt; 73  A_woodhouseii_334142  0.8            29</span>
<span class="co">#&gt; 75  A_woodhouseii_334153  0.8            29</span>
<span class="co">#&gt; 76 A_coerulescens_396256  0.8            29</span>
<span class="co">#&gt; 80 A_coerulescens_396264  0.8            29</span>

<span class="co">#drop the three samples with an excess of missing data at an 80% SNP completeness threshold</span>
<span class="va">vcfR</span><span class="op">&lt;-</span> <span class="va">vcfR</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"A_coerulescens_396263"</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"A_woodhouseii_334134"</span> <span class="op">&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"A_coerulescens_396256"</span><span class="op">]</span>

<span class="co">#remove invariant SNPs</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span><span class="va">vcfR</span>, min.mac <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>
<span class="co">#&gt; 3.54% of SNPs fell below a minor allele count of 1 and were removed from the VCF</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-9-1.png" width="384"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vcfR</span>
<span class="co">#&gt; ***** Object of Class vcfR *****</span>
<span class="co">#&gt; 14 samples</span>
<span class="co">#&gt; 1 CHROMs</span>
<span class="co">#&gt; 191 variants</span>
<span class="co">#&gt; Object size: 0.2 Mb</span>
<span class="co">#&gt; 31.71 percent missing data</span>
<span class="co">#&gt; *****        *****         *****</span>

<span class="co">#update popmap by removing samples that have been filtered out</span>
<span class="va">popmap</span><span class="op">&lt;-</span><span class="va">popmap</span><span class="op">[</span><span class="va">popmap</span><span class="op">$</span><span class="va">id</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,<span class="op">]</span></code></pre></div>
<p>Re-visualize missing data by SNP and the effect of various cutoffs on the missingness of each sample and set a reasonable missing data cutoff.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#visualize missing data at various completeness thresholds</span>
<span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span>
<span class="co">#&gt; cutoff is not specified, exploratory visualizations will be generated</span>
<span class="co">#&gt; Picking joint bandwidth of 0.124</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-10-1.png" width="384"><img src="reproducible-vignette_files/figure-html/unnamed-chunk-10-2.png" width="384"></p>
<pre><code>#&gt;    filt missingness snps.retained
#&gt; 1  0.30  0.20506912           155
#&gt; 2  0.50  0.15631470           138
#&gt; 3  0.60  0.10593220           118
#&gt; 4  0.65  0.08010681           107
#&gt; 5  0.70  0.08010681           107
#&gt; 6  0.75  0.05654762            96
#&gt; 7  0.80  0.04414125            89
#&gt; 8  0.85  0.04414125            89
#&gt; 9  0.90  0.02250489            73
#&gt; 10 0.95  0.00000000            50
#&gt; 11 1.00  0.00000000            50
#all samples look good at most thresholds, because of the small size of this dataset, I will choose a 60% completeness threshold in order to retain as many SNPs as possible

#filter vcfR
vcfR&lt;-missing_by_snp(vcfR, cutoff = .6)
#&gt; cutoff is specified, filtered vcfR object will be returned
#&gt; 38.22% of SNPs fell below a completeness cutoff of 0.6 and were removed from the VCF</code></pre>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-10-3.png" width="384"></p>
</div>
</div>
<div id="step-5-quality-unaware-filters" class="section level1">
<h1 class="hasAnchor">
<a href="#step-5-quality-unaware-filters" class="anchor"></a>Step 5: quality unaware filters</h1>
<p>We can now implement filters that are blind to genotype quality like a Minor Allele Count (MAC) threshold, and a distance-based filtering threshold, both of which may serve to increase the signal to noise ratio of our dataset in downstream analyses.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#remove singletons (loci with only a single variant allele which have no phylogenetic signal)</span>
<span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">vcfR</span>, min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>
<span class="co">#&gt; 38.14% of SNPs fell below a minor allele count of 2 and were removed from the VCF</span></code></pre></div>
<p><img src="reproducible-vignette_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="co">#linkage filter vcf to thin SNPs to one per 500bp</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>vcfR<span class="ot">&lt;-</span><span class="fu">distance_thin</span>(vcfR, <span class="at">min.distance =</span> <span class="dv">500</span>)</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 61 out of 73 input SNPs were not located within 500 base-pairs of another SNP and were retained despite filtering</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a><span class="co">#look at final stats for our filtered vcf file</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>vcfR</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14 samples</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 CHROMs</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 61 variants</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Object size: 0.1 Mb</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12.76 percent missing data</span></span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; *****        *****         *****</span></span></code></pre></div>
</div>
<div id="step-6-write-out-files-for-downstream-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#step-6-write-out-files-for-downstream-analysis" class="anchor"></a>Step 6: write out files for downstream analysis</h1>
<p>We can now use the convenient function vcfR::write.vcf() to export our filtered vcf file for downstream analyses</p>
<div id="note-3" class="section level3">
<h3 class="hasAnchor">
<a href="#note-3" class="anchor"></a>Note:</h3>
<p>the function vcfR::write.vcf() automatically writes a gzipped vcf file, so be sure to add the suffix .gz to the name of your output file.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#write out vcf</span>
<span class="co">#vcfR::write.vcf(vcfR, file = "~/Downloads/scrub.jay.example.filtered.vcf.gz")</span></code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
