<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>scrub-jay-RADseq-vignette • SNPfiltR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="scrub-jay-RADseq-vignette">
<meta property="og:description" content="SNPfiltR">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/accuracy-benchmarking.html">accuracy-benchmarking</a>
    </li>
    <li>
      <a href="../articles/performance-benchmarking.html">Performance benchmarking</a>
    </li>
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-RADseq-vignette.html">scrub-jay-RADseq-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-UCE-vignette.html">scrub-jay-UCE-vignette</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>scrub-jay-RADseq-vignette</h1>
            
      
      
      <div class="hidden name"><code>scrub-jay-RADseq-vignette.Rmd</code></div>

    </div>

    
    
<p>This is a real-world example of using the SNPfiltR package to filter
a moderately sized RADseq dataset. This dataset includes 115 samples,
and &gt; 200,000 unfiltered SNPs. Reading this unfiltered vcf file into
Rstudio took ~1 minute on my personal laptop, and individual functions
took between a few seconds and a few minutes to finish. Overall, this
dataset is a great example of the power of R and the SNPfiltR package to
streamline the filtering process for moderately sized
reduced-representation genomic datasets.</p>
<div class="section level2">
<h2 id="optional-step-0-per-sample-quality-control">Optional Step 0: per sample quality control<a class="anchor" aria-label="anchor" href="#optional-step-0-per-sample-quality-control"></a>
</h2>
<p>Do quality control per sample before performing SNP calling. I have
written an <a href="https://github.com/DevonDeRaad/RADstackshelpR/blob/master/inst/extdata/fastqcr.Rmd" class="external-link">RMarkdown
script</a> that uses the R package <a href="https://github.com/kassambara/fastqcr" class="external-link">fastqcr</a> to generate a
report visualizing the quality and quantity of sequencing for each
sample, and recommending a subset of samples to be immediately dropped
before parameter optimization (specifically useful for RADseq data). The
only modification necessary for this script is the path to the folder
containing the input .fastq.gz files and the path to your desired output
folder. An example report generated using this script can be seen <a href="https://devonderaad.github.io/RADstackshelpR/articles/quality.control.vignette.html" class="external-link">here</a>.
Because the fastq.gz files for your experiment may be large and handled
remotely, an example bash script for executing this RMarkdown file as a
job on a high performance computing cluster is available <a href="https://github.com/DevonDeRaad/RADstackshelpR/blob/master/inst/extdata/RMarkdown.qc.submit.script.sh" class="external-link">here</a>.</p>
<div class="section level5">
<h5 id="because-this-dataset-was-not-overly-large-and-i-used-a-reference-based-assembly-that-doesnt-depend-on-each-sample-contributing-to-the-de-novo-building-of-rad-loci-i-chose-to-skip-this-step-and-do-all-of-the-per-sample-quality-filtering-in-r--i-started-below-by-reading-in-the-vcf-file-using-the-aforementioned-vcfr-package">Because this dataset was not overly large, and I used a reference
based assembly that doesn’t depend on each sample contributing to the de
novo building of RAD loci, I chose to skip this step and do all of the
per sample quality filtering in R. I started below by reading in the vcf
file using the aforementioned vcfR package:<a class="anchor" aria-label="anchor" href="#because-this-dataset-was-not-overly-large-and-i-used-a-reference-based-assembly-that-doesnt-depend-on-each-sample-contributing-to-the-de-novo-building-of-rad-loci-i-chose-to-skip-this-step-and-do-all-of-the-per-sample-quality-filtering-in-r--i-started-below-by-reading-in-the-vcf-file-using-the-aforementioned-vcfr-package"></a>
</h5>
</div>
</div>
<div class="section level2">
<h2 id="step-1-read-in-vcf-file-as-vcfr-object">Step 1: read in vcf file as ‘vcfR’ object<a class="anchor" aria-label="anchor" href="#step-1-read-in-vcf-file-as-vcfr-object"></a>
</h2>
<div class="section level4">
<h4 id="read-in-vcf-file-using-vcfr">Read in vcf file using <a href="https://knausb.github.io/vcfR_documentation/" class="external-link">vcfR</a><a class="anchor" aria-label="anchor" href="#read-in-vcf-file-using-vcfr"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SNPfiltR</span><span class="op">)</span></span>
<span><span class="co">#&gt; This is SNPfiltR v.1.0.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Detailed usage information is available at: devonderaad.github.io/SNPfiltR/ </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; If you use SNPfiltR in your published work, please cite the following papers: </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; DeRaad, D.A. (2022), SNPfiltR: an R package for interactive and reproducible SNP filtering. Molecular Ecology Resources, 00, 1-15. http://doi.org/10.1111/1755-0998.13618 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Knaus, Brian J., and Niklaus J. Grunwald. 2017. VCFR: a package to manipulate and visualize variant call format data in R. Molecular Ecology Resources, 17.1:44-53. http://doi.org/10.1111/1755-0998.12549</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR" class="external-link">vcfR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    *****       ***   vcfR   ***       *****</span></span>
<span><span class="co">#&gt;    This is vcfR 1.14.0 </span></span>
<span><span class="co">#&gt;      browseVignettes('vcfR') # Documentation</span></span>
<span><span class="co">#&gt;      citation('vcfR') # Citation</span></span>
<span><span class="co">#&gt;    *****       *****      *****       *****</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#read in vcf as vcfR</span></span>
<span><span class="va">vcfR</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/io_vcfR.html" class="external-link">read.vcfR</a></span><span class="op">(</span><span class="st">"~/Desktop/aph.data/populations.snps.vcf"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### check the metadata present in your vcf</span></span>
<span><span class="va">vcfR</span></span>
<span><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span><span class="co">#&gt; 115 samples</span></span>
<span><span class="co">#&gt; 87 CHROMs</span></span>
<span><span class="co">#&gt; 210,336 variants</span></span>
<span><span class="co">#&gt; Object size: 685.5 Mb</span></span>
<span><span class="co">#&gt; 57.1 percent missing data</span></span>
<span><span class="co">#&gt; *****        *****         *****</span></span>
<span></span>
<span><span class="co">#generate popmap file. Two column popmap with the same format as stacks, and the columns must be named 'id' and 'pop'</span></span>
<span><span class="va">popmap</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>,pop<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>, <span class="fl">3</span>,<span class="fl">11</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-2-quality-filtering">Step 2: quality filtering<a class="anchor" aria-label="anchor" href="#step-2-quality-filtering"></a>
</h2>
<div class="section level4">
<h4 id="implement-quality-filters-that-dont-involve-missing-data--this-is-because-removing-low-data-samples-will-alter-percentagequantile-based-missing-data-cutoffs-so-we-wait-to-implement-those-until-after-deciding-on-our-final-set-of-samples-for-downstream-analysis">Implement quality filters that don’t involve missing data. This is
because removing low data samples will alter percentage/quantile based
missing data cutoffs, so we wait to implement those until after deciding
on our final set of samples for downstream analysis<a class="anchor" aria-label="anchor" href="#implement-quality-filters-that-dont-involve-missing-data--this-is-because-removing-low-data-samples-will-alter-percentagequantile-based-missing-data-cutoffs-so-we-wait-to-implement-those-until-after-deciding-on-our-final-set-of-samples-for-downstream-analysis"></a>
</h4>
</div>
<div class="section level4">
<h4 id="note">Note:<a class="anchor" aria-label="anchor" href="#note"></a>
</h4>
<p>I like to start with a completely unfiltered vcf file, in order to
implement a cohesive, fully documentable filtering pipeline in R.
However, if I have a very large unfiltered vcf file that I would like to
work with in Rstudio, I sometimes perform hard filtering, or implement
some conservative percentage based filters (e.g., remove all SNPs above
90% missing data) via command-line calls to the program VCFtools, in
order to reduce the size of the starting vcf that I need to read into
Rstudio. Just be aware that once you drop low data samples, your
previously enforced (per SNP or locus) missing data % will no longer be
accurate. VCFtools has faster implementations of some of the same
filters implemented here, and really shines with large datasets. The
R-based implementations of these filters in SNPfiltR offers interactive
visualization in a cohesive R-based pipeline, but be aware that reading
in excessively large files to Rstudio (typically, &gt; 1 Gb in my
experience) can cause R to hang and crash. Luckily, I have found that
variant sites only vcf files for most reduced-representation genomic
datasets are under this size, and can be handled efficiently using
Rstudio on a personal computer.</p>
</div>
<div class="section level4">
<h4 id="note-1">Note:<a class="anchor" aria-label="anchor" href="#note-1"></a>
</h4>
<p>Jon Puritz has an excellent filtering tutorial that is focused
specifically on <a href="https://www.ddocent.com/filtering/" class="external-link">filtering
RADseq data</a> Multiple functions in SNPfiltR were generated in order
to follow the guidelines and suggestions laid out in this tutorial. We
can follow these guidelines for hard filtering (he suggests minimum
depth=3, gq =30), and can implement suggested filters like allele
balance and max depth, here in R using SNPfiltR.</p>
<div class="section level5">
<h5 id="start-by-visualizing-the-distributions-of-depth-of-sequencing-and-genotype-quality-among-called-genotypes-then-set-appropriate-cutoffs-for-both-values-for-this-dataset-">start by visualizing the distributions of depth of sequencing and
genotype quality among called genotypes, then set appropriate cutoffs
for both values for this dataset.<a class="anchor" aria-label="anchor" href="#start-by-visualizing-the-distributions-of-depth-of-sequencing-and-genotype-quality-among-called-genotypes-then-set-appropriate-cutoffs-for-both-values-for-this-dataset-"></a>
</h5>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#visualize distributions</span></span>
<span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span><span class="op">)</span></span>
<span><span class="co">#&gt; no depth cutoff provided, exploratory visualization will be generated.</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-5-1.png" width="900"></p>
<pre><code><span><span class="co">#&gt; no genotype quality cutoff provided, exploratory visualization will be generated.</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-5-2.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-5-3.png" width="900"></p>
<pre><code><span><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span><span class="co">#&gt; 115 samples</span></span>
<span><span class="co">#&gt; 87 CHROMs</span></span>
<span><span class="co">#&gt; 210,336 variants</span></span>
<span><span class="co">#&gt; Object size: 685.5 Mb</span></span>
<span><span class="co">#&gt; 57.1 percent missing data</span></span>
<span><span class="co">#&gt; *****        *****         *****</span></span>
<span></span>
<span><span class="co">#hard filter to minimum depth of 5, and minimum genotype quality of 30</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/hard_filter.html">hard_filter</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, depth <span class="op">=</span> <span class="fl">5</span>, gq <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span><span class="co">#&gt; 32.92% of genotypes fall below a read depth of 5 and were converted to NA</span></span>
<span><span class="co">#&gt; 2.01% of genotypes fall below a genotype quality of 30 and were converted to NA</span></span>
<span></span>
<span><span class="co">#remove loci with &gt; 2 alleles</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/filter_biallelic.html">filter_biallelic</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span></span>
<span><span class="co">#&gt; 0 SNPs, 0% of all input SNPs, contained more than 2 alleles, and were removed from the VCF</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-5-4.png" width="900"></p>
</div>
<div class="section level5">
<h5 id="then-use-this-function-to-filter-for-allele-balance">Then use this function to filter for allele balance<a class="anchor" aria-label="anchor" href="#then-use-this-function-to-filter-for-allele-balance"></a>
</h5>
<p>From the <a href="https://www.ddocent.com/filtering/" class="external-link">Ddocent SNP
filtering tutorial</a> “Allele balance: a number between 0 and 1
representing the ratio of reads showing the reference allele to all
reads, considering only reads from individuals called as heterozygous,
we expect that the allele balance in our data (for real loci) should be
close to 0.5”</p>
</div>
<div class="section level5">
<h5 id="the-snpfiltr-allele-balance-function-will-convert-heterozygous-genotypes-to-missing-if-they-fall-outside-of-the--25--75-range-">the SNPfiltR allele balance function will convert heterozygous
genotypes to missing if they fall outside of the .25-.75 range.<a class="anchor" aria-label="anchor" href="#the-snpfiltr-allele-balance-function-will-convert-heterozygous-genotypes-to-missing-if-they-fall-outside-of-the--25--75-range-"></a>
</h5>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#execute allele balance filter</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/filter_allele_balance.html">filter_allele_balance</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span></span>
<span><span class="co">#&gt; 7.56% of het genotypes (0.39% of all genotypes) fall outside of 0.25 - 0.75 allele balance ratio and were converted to NA</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-6-1.png" width="900"></p>
<p>Now we can execute a max depth filter (super high depth loci are
likely multiple loci stuck together into a single paralogous locus).</p>
</div>
</div>
<div class="section level4">
<h4 id="note-2">Note:<a class="anchor" aria-label="anchor" href="#note-2"></a>
</h4>
<p>This filter is applied ‘per SNP’ rather than ‘per genotype’ otherwise
we would simply be removing most of the genotypes from our deepest
sequenced samples (because sequencing depth is so variable between
samples). By filtering per SNP, we remove the SNPs with outlier depth
values, which are most likely to be spuriously mapped/built paralagous
loci.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#visualize and pick appropriate max depth cutoff</span></span>
<span><span class="fu"><a href="../reference/max_depth.html">max_depth</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span></span>
<span><span class="co">#&gt; cutoff is not specified, exploratory visualization will be generated.</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-7-1.png" width="900"></p>
<pre><code><span><span class="co">#&gt; dashed line indicates a mean depth across all SNPs of 46.7</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-7-2.png" width="900"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#filter vcf by the max depth cutoff you chose</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/max_depth.html">max_depth</a></span><span class="op">(</span><span class="va">vcfR</span>, maxdepth <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; maxdepth cutoff is specified, filtered vcfR object will be returned</span></span>
<span><span class="co">#&gt; 12.85% of SNPs were above a mean depth of 100 and were removed from the vcf</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-7-3.png" width="900"></p>
</div>
<div class="section level4">
<h4 id="note-3">Note:<a class="anchor" aria-label="anchor" href="#note-3"></a>
</h4>
<p>It may be a good idea to additionally filter out SNPs that are
significantly out of HWE if you have a really good idea of what the
population structure in your sample looks like and good sample sizes in
each pop. For this dataset, which is highly structured (many isolated
island pops) with species boundaries that are in flux, I am not going to
use a HWE filter, because I don’t feel comfortable confidently
identifying populations in which we can expect HWE. Many other programs
(such as VCFtools) can filter according to HWE if desired.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#remove invariant SNPs generated during the genotype filtering steps</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span><span class="va">vcfR</span>, min.mac <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; 55.87% of SNPs fell below a minor allele count of 1 and were removed from the VCF</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#check vcfR to see how many SNPs we have left</span></span>
<span><span class="va">vcfR</span></span>
<span><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span><span class="co">#&gt; 115 samples</span></span>
<span><span class="co">#&gt; 70 CHROMs</span></span>
<span><span class="co">#&gt; 80,885 variants</span></span>
<span><span class="co">#&gt; Object size: 283.2 Mb</span></span>
<span><span class="co">#&gt; 60.85 percent missing data</span></span>
<span><span class="co">#&gt; *****        *****         *****</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-3-set-missing-data-per-sample-cutoff">Step 3: set missing data per sample cutoff<a class="anchor" aria-label="anchor" href="#step-3-set-missing-data-per-sample-cutoff"></a>
</h2>
<div class="section level4">
<h4 id="set-arbitrary-cutoff-for-missing-data-allowed-per-sample-">Set arbitrary cutoff for missing data allowed per sample.<a class="anchor" aria-label="anchor" href="#set-arbitrary-cutoff-for-missing-data-allowed-per-sample-"></a>
</h4>
<p>Determining which samples and SNPs to retain is always project
specific, and is contingent on sampling, biology of the focal taxa,
sequencing idiosyncrasies, etc. SNPfiltR contains functions designed to
simply and easily generate exploratory visualizations that will allow
you to make informed decisions about which samples and SNPs are of
sufficient quality to retain for downstream analyses, but there is never
a single correct option for these cutoffs. In my experience, the best
thing to do is to look at your data, look at what effects some
reasonable cutoffs would have on your data, and pick one that works for
you. Then as you continue to analyze your data, make sure that your
arbitrary filtering decisions are not driving the patterns you see, and
iteratively update your filtering approach if you are concerned about
the effects previous filtering choices are having on downstream
results.</p>
</div>
<div class="section level4">
<h4 id="we-will-start-by-determining-which-samples-contain-too-few-sequences-to-be-used-in-downstream-analyses-by-visualizing-missing-data-per-sample">We will start by determining which samples contain too few sequences
to be used in downstream analyses, by visualizing missing data per
sample<a class="anchor" aria-label="anchor" href="#we-will-start-by-determining-which-samples-contain-too-few-sequences-to-be-used-in-downstream-analyses-by-visualizing-missing-data-per-sample"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#run function to visualize samples</span></span>
<span><span class="fu"><a href="../reference/missing_by_sample.html">missing_by_sample</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, popmap <span class="op">=</span> <span class="va">popmap</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bin width defaults to 1/30 of the range of the data. Pick better value with</span></span>
<span><span class="co">#&gt; `binwidth`.</span></span>
<span><span class="co">#&gt; Bin width defaults to 1/30 of the range of the data. Pick better value with</span></span>
<span><span class="co">#&gt; `binwidth`.</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-9-1.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-9-2.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-9-3.png" width="900"></p>
</div>
<div class="section level4">
<h4 id="now-we-can-try-setting-a-reasonable-threshold-and-then-visualizing-clustering-patterns-to-determine-whether-there-are-still-problematic-samples-with-excess-missing-data-that-cant-be-accurately-assigned-to-genetic-clusters">Now we can try setting a reasonable threshold, and then visualizing
clustering patterns to determine whether there are still problematic
samples with excess missing data that can’t be accurately assigned to
genetic clusters<a class="anchor" aria-label="anchor" href="#now-we-can-try-setting-a-reasonable-threshold-and-then-visualizing-clustering-patterns-to-determine-whether-there-are-still-problematic-samples-with-excess-missing-data-that-cant-be-accurately-assigned-to-genetic-clusters"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#run function to drop samples above the threshold we want from the vcf</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_sample.html">missing_by_sample</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, cutoff <span class="op">=</span> <span class="fl">.81</span><span class="op">)</span></span>
<span><span class="co">#&gt; 20 samples are above a 0.81 missing data cutoff, and were removed from VCF</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-10-1.png" width="900"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#subset popmap to only include retained individuals</span></span>
<span><span class="va">popmap</span><span class="op">&lt;-</span><span class="va">popmap</span><span class="op">[</span><span class="va">popmap</span><span class="op">$</span><span class="va">id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">@</span><span class="va">gt</span><span class="op">)</span>,<span class="op">]</span></span>
<span></span>
<span><span class="co">#remove invariant sites generated by dropping individuals</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span><span class="va">vcfR</span>, min.mac <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="co">#&gt; 0.61% of SNPs fell below a minor allele count of 1 and were removed from the VCF</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-10-2.png" width="900"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#verify that missing data is not driving clustering patterns among the retained samples</span></span>
<span><span class="va">miss</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/assess_missing_data_pca.html">assess_missing_data_pca</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, popmap <span class="op">=</span> <span class="va">popmap</span>, thresholds <span class="op">=</span> <span class="fl">.8</span>, clustering <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span>
<span><span class="co">#&gt; 76.56% of SNPs fell below a completeness cutoff of 0.8 and were removed from the VCF</span></span>
<span><span class="co">#&gt; Loading required namespace: adegenet</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-10-3.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-10-4.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-10-5.png" width="900"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#if there are still problematic samples, drop them using the following syntax</span></span>
<span><span class="co">#vcfR &lt;- vcfR[,colnames(vcfR@gt) != "A_woodhouseii_24711" &amp; colnames(vcfR@gt) != "A_californica_45901"]</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-4-set-missing-data-per-snp-cutoff">Step 4: set missing data per SNP cutoff<a class="anchor" aria-label="anchor" href="#step-4-set-missing-data-per-snp-cutoff"></a>
</h2>
<div class="section level4">
<h4 id="set-arbitrary-cutoff-for-missing-data-allowed-per-snp-">Set arbitrary cutoff for missing data allowed per SNP.<a class="anchor" aria-label="anchor" href="#set-arbitrary-cutoff-for-missing-data-allowed-per-snp-"></a>
</h4>
<div class="section level5">
<h5 id="we-can-visualize-the-effect-that-typical-missing-data-cutoffs-will-have-on-both-the-number-of-snps-retained-and-the-total-missing-data-in-our-entire-dataset-we-want-to-choose-a-cutoff-that-minimizes-the-overall-missing-data-in-the-dataset-while-maximizing-the-total-number-of-loci-retained-">We can visualize the effect that typical missing data cutoffs will
have on both the number of SNPs retained and the total missing data in
our entire dataset.We want to choose a cutoff that minimizes the overall
missing data in the dataset, while maximizing the total number of loci
retained.<a class="anchor" aria-label="anchor" href="#we-can-visualize-the-effect-that-typical-missing-data-cutoffs-will-have-on-both-the-number-of-snps-retained-and-the-total-missing-data-in-our-entire-dataset-we-want-to-choose-a-cutoff-that-minimizes-the-overall-missing-data-in-the-dataset-while-maximizing-the-total-number-of-loci-retained-"></a>
</h5>
</div>
</div>
<div class="section level4">
<h4 id="note-4">Note:<a class="anchor" aria-label="anchor" href="#note-4"></a>
</h4>
<p>This filter interacts with the above filter, where we dropped low
data samples. A good rule of thumb is that individual samples shouldn’t
be above 50% missing data after applying a per-SNP missing data cutoff.
So if we are retaining specific low data samples out of necessity or
project design, we may have to set a more stringent per-SNP missing data
cutoff, at the expense of the total number of SNPs retained for
downstream analyses. We can again use the assess_missing_data_pca()
function to determine whether all retained samples contain enough data
at our chosen cutoff in order to be assigned accurately to their species
group.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#visualize missing data by SNP and the effect of various cutoffs on the missingness of each sample</span></span>
<span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span><span class="va">vcfR</span><span class="op">)</span></span>
<span><span class="co">#&gt; cutoff is not specified, exploratory visualizations will be generated</span></span>
<span><span class="co">#&gt; Picking joint bandwidth of 0.0645</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-1.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-2.png" width="900"></p>
<pre><code><span><span class="co">#&gt;    filt missingness snps.retained</span></span>
<span><span class="co">#&gt; 1  0.30  0.30436210         47570</span></span>
<span><span class="co">#&gt; 2  0.50  0.20138396         35398</span></span>
<span><span class="co">#&gt; 3  0.60  0.15772868         30174</span></span>
<span><span class="co">#&gt; 4  0.65  0.13437846         27298</span></span>
<span><span class="co">#&gt; 5  0.70  0.11387863         24657</span></span>
<span><span class="co">#&gt; 6  0.75  0.09326028         21836</span></span>
<span><span class="co">#&gt; 7  0.80  0.07731504         19491</span></span>
<span><span class="co">#&gt; 8  0.85  0.05743029         16307</span></span>
<span><span class="co">#&gt; 9  0.90  0.03871199         12790</span></span>
<span><span class="co">#&gt; 10 0.95  0.01867360          8013</span></span>
<span><span class="co">#&gt; 11 1.00  0.00000000          1779</span></span>
<span></span>
<span><span class="co">#check if these cutoffs prevent samples from clustering</span></span>
<span><span class="co">#by patterns of missing data in a PCA</span></span>
<span><span class="va">miss</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/assess_missing_data_pca.html">assess_missing_data_pca</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, popmap <span class="op">=</span> <span class="va">popmap</span>,</span>
<span>                              thresholds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.75</span>,<span class="fl">.85</span>,<span class="fl">.95</span><span class="op">)</span>, clustering <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span>
<span><span class="co">#&gt; 72.84% of SNPs fell below a completeness cutoff of 0.75 and were removed from the VCF</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-3.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-4.png" width="900"></p>
<pre><code><span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-5.png" width="900"></p>
<pre><code><span><span class="co">#&gt; 79.72% of SNPs fell below a completeness cutoff of 0.85 and were removed from the VCF</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-6.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-7.png" width="900"></p>
<pre><code><span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-8.png" width="900"></p>
<pre><code><span><span class="co">#&gt; 90.03% of SNPs fell below a completeness cutoff of 0.95 and were removed from the VCF</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-9.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-10.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-11.png" width="900"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#check what t-SNE clustering looks like at an 85% threshold</span></span>
<span><span class="va">miss</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/assess_missing_data_tsne.html">assess_missing_data_tsne</a></span><span class="op">(</span>vcfR<span class="op">=</span><span class="va">vcfR</span>, popmap <span class="op">=</span> <span class="va">popmap</span>,</span>
<span>                              thresholds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">.75</span>,<span class="fl">.85</span>,<span class="fl">.95</span><span class="op">)</span>, clustering <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span>
<span><span class="co">#&gt; 72.84% of SNPs fell below a completeness cutoff of 0.75 and were removed from the VCF</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-12.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-13.png" width="900"></p>
<pre><code><span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-14.png" width="900"></p>
<pre><code><span><span class="co">#&gt; 79.72% of SNPs fell below a completeness cutoff of 0.85 and were removed from the VCF</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-15.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-16.png" width="900"></p>
<pre><code><span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-17.png" width="900"></p>
<pre><code><span><span class="co">#&gt; 90.03% of SNPs fell below a completeness cutoff of 0.95 and were removed from the VCF</span></span></code></pre>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-18.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-19.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-20.png" width="900"></p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#choose a cutoff resulting in an acceptable amount of missing data in each sample, and maximizes SNPs retained while minimizing overall missing data, and filter vcf</span></span>
<span><span class="va">vcfR</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/missing_by_snp.html">missing_by_snp</a></span><span class="op">(</span><span class="va">vcfR</span>, cutoff <span class="op">=</span> <span class="fl">.85</span><span class="op">)</span></span>
<span><span class="co">#&gt; cutoff is specified, filtered vcfR object will be returned</span></span>
<span><span class="co">#&gt; 79.72% of SNPs fell below a completeness cutoff of 0.85 and were removed from the VCF</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-11-21.png" width="900"></p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#check how many SNPs and samples are left</span></span>
<span><span class="va">vcfR</span></span>
<span><span class="co">#&gt; ***** Object of Class vcfR *****</span></span>
<span><span class="co">#&gt; 95 samples</span></span>
<span><span class="co">#&gt; 36 CHROMs</span></span>
<span><span class="co">#&gt; 16,307 variants</span></span>
<span><span class="co">#&gt; Object size: 111.2 Mb</span></span>
<span><span class="co">#&gt; 5.743 percent missing data</span></span>
<span><span class="co">#&gt; *****        *****         *****</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-5-minor-allele-and-linkage-filtering">Step 5: minor allele and linkage filtering<a class="anchor" aria-label="anchor" href="#step-5-minor-allele-and-linkage-filtering"></a>
</h2>
<div class="section level4">
<h4 id="investigate-the-effect-of-a-minor-allele-count-mac-cutoff-on-downstream-inferences--we-always-want-to-do-this-last-because-these-filters-are-not-quality-aware-and-using-them-early-in-the-pipeline-will-result-in-dumping-good-snps-">investigate the effect of a minor allele count (MAC) cutoff on
downstream inferences. We always want to do this last, because these
filters are not quality aware, and using them early in the pipeline will
result in dumping good SNPs.<a class="anchor" aria-label="anchor" href="#investigate-the-effect-of-a-minor-allele-count-mac-cutoff-on-downstream-inferences--we-always-want-to-do-this-last-because-these-filters-are-not-quality-aware-and-using-them-early-in-the-pipeline-will-result-in-dumping-good-snps-"></a>
</h4>
<div class="section level5">
<h5 id="macmaf-cutoffs-can-be-helpful-in-removing-spurious-and-uninformative-loci-from-the-dataset-but-also-have-the-potential-to-bias-downstream-inferences--linck-and-battey-2019-have-an-excellent-paper-on-just-this-topic--from-the-paper-">MAC/MAF cutoffs can be helpful in removing spurious and
uninformative loci from the dataset, but also have the potential to bias
downstream inferences. Linck and Battey (2019) have an excellent paper
on just this topic. From the paper-<a class="anchor" aria-label="anchor" href="#macmaf-cutoffs-can-be-helpful-in-removing-spurious-and-uninformative-loci-from-the-dataset-but-also-have-the-potential-to-bias-downstream-inferences--linck-and-battey-2019-have-an-excellent-paper-on-just-this-topic--from-the-paper-"></a>
</h5>
<p>“We recommend researchers using model‐based programs to describe
population structure observe the following best practices: (a) duplicate
analyses with nonparametric methods suchas PCA and DAPC with cross
validation (b) exclude singletons (c) compare alignments with multiple
assembly parameters When seeking to exclude only singletons in
alignments with missing data (a ubiquitous problem for
reduced‐representation library preparation methods), it is preferable to
filter by the count (rather than frequency) of the minor allele, because
variation in the amount of missing data across an alignment will cause a
static frequency cutoff to remove different SFS classes at different
sites”</p>
</div>
<div class="section level5">
<h5 id="our-package-contains-a-convenient-wrapper-functions-that-can-filter-based-on-minor-allele-count-mac-and-streamline-investigation-of-the-effects-of-various-filtering-parameters-on-sample-clustering-patterns-">Our package contains a convenient wrapper functions that can filter
based on minor allele count (MAC) and streamline investigation of the
effects of various filtering parameters on sample clustering
patterns.<a class="anchor" aria-label="anchor" href="#our-package-contains-a-convenient-wrapper-functions-that-can-filter-based-on-minor-allele-count-mac-and-streamline-investigation-of-the-effects-of-various-filtering-parameters-on-sample-clustering-patterns-"></a>
</h5>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#investigate clustering patterns with and without a minor allele cutoff</span></span>
<span><span class="co">#use min.mac() to investigate the effect of multiple cutoffs</span></span>
<span><span class="va">vcfR.mac</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/min_mac.html">min_mac</a></span><span class="op">(</span>vcfR <span class="op">=</span> <span class="va">vcfR</span>, min.mac <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; 30.87% of SNPs fell below a minor allele count of 2 and were removed from the VCF</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-12-1.png" width="900"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#assess clustering without MAC cutoff</span></span>
<span><span class="va">miss</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/assess_missing_data_tsne.html">assess_missing_data_tsne</a></span><span class="op">(</span><span class="va">vcfR</span>, <span class="va">popmap</span>, clustering <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-12-2.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-12-3.png" width="900"></p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co">#assess clustering with MAC cutoff</span></span>
<span><span class="va">miss</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/assess_missing_data_tsne.html">assess_missing_data_tsne</a></span><span class="op">(</span><span class="va">vcfR.mac</span>, <span class="va">popmap</span>, clustering <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-12-4.png" width="900"><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-12-5.png" width="900"></p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co">#based on these visualizations, singletons are not obviously biasing clustering patterns, so I will leave them in for now. If I want to run a program like STRUCTURE, where singletons are known to bias inference, I can write out the vcf with singletons removed as well:</span></span>
<span><span class="co">#vcfR::write.vcf(vcfR.mac, file = "~/Downloads/scrub.jay.mac.vcf.gz")</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="finally-we-will-make-sure-that-the-depth-and-genotype-quality-look-consistent-across-snps-and-samples-following-our-filtering-pipeline-">Finally, we will make sure that the depth and genotype quality look
consistent across SNPs and samples, following our filtering
pipeline.<a class="anchor" aria-label="anchor" href="#finally-we-will-make-sure-that-the-depth-and-genotype-quality-look-consistent-across-snps-and-samples-following-our-filtering-pipeline-"></a>
</h5>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#plot depth per snp and per sample</span></span>
<span><span class="va">dp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/extract_gt.html" class="external-link">extract.gt</a></span><span class="op">(</span><span class="va">vcfR</span>, element <span class="op">=</span> <span class="st">"DP"</span>, as.numeric<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/heatmap_bp.html" class="external-link">heatmap.bp</a></span><span class="op">(</span><span class="va">dp</span>, rlabels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-13-1.png" width="900"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#plot genotype quality per snp and per sample</span></span>
<span><span class="va">gq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/extract_gt.html" class="external-link">extract.gt</a></span><span class="op">(</span><span class="va">vcfR</span>, element <span class="op">=</span> <span class="st">"GQ"</span>, as.numeric<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/heatmap_bp.html" class="external-link">heatmap.bp</a></span><span class="op">(</span><span class="va">gq</span>, rlabels <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="scrub-jay-RADseq-vignette_files/figure-html/unnamed-chunk-13-2.png" width="900"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="step-6-write-out-vcf-files-for-downstream-analyses-">Step 6: write out vcf files for downstream analyses.<a class="anchor" aria-label="anchor" href="#step-6-write-out-vcf-files-for-downstream-analyses-"></a>
</h2>
<div class="section level5">
<h5 id="optionally-we-can-use-the-distance_thin-function-from-the-snpfiltr-package-in-order-to-filter-our-snps-to-a-minimum-distance-between-snps-in-order-to-get-a-set-of-unlinked-snps-for-downsteam-analyses">Optionally, we can use the distance_thin() function from the
SNPfiltR package in order to filter our SNPs to a minimum distance
between SNPs, in order to get a set of unlinked SNPs for downsteam
analyses<a class="anchor" aria-label="anchor" href="#optionally-we-can-use-the-distance_thin-function-from-the-snpfiltr-package-in-order-to-filter-our-snps-to-a-minimum-distance-between-snps-in-order-to-get-a-set-of-unlinked-snps-for-downsteam-analyses"></a>
</h5>
</div>
<div class="section level4">
<h4 id="note-5">Note:<a class="anchor" aria-label="anchor" href="#note-5"></a>
</h4>
<p>The function vcfR::write.vcf() automatically writes a gzipped vcf
file, so be sure to add the suffix .gz to the name of your output
file.</p>
<div class="section level5">
<h5 id="write-out-the-filtered-vcf-and-a-linkage-filtered-version-for-downstream-analyses">Write out the filtered vcf, and a linkage filtered version, for
downstream analyses<a class="anchor" aria-label="anchor" href="#write-out-the-filtered-vcf-and-a-linkage-filtered-version-for-downstream-analyses"></a>
</h5>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co">#fix mislabeled sample</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(vcfR<span class="sc">@</span>gt)[<span class="fu">colnames</span>(vcfR<span class="sc">@</span>gt) <span class="sc">==</span> <span class="st">"A_californica_334171"</span>]<span class="ot">&lt;-</span><span class="st">"A_woodhouseii_334171"</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="co">#write out vcf with all SNPs</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>vcfR<span class="sc">::</span><span class="fu">write.vcf</span>(vcfR, <span class="st">"~/Downloads/aphelocoma.filtered.vcf.gz"</span>)</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a><span class="co">#linkage filter vcf to thin SNPs to one per 500bp</span></span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>vcfR.thin<span class="ot">&lt;-</span><span class="fu">distance_thin</span>(vcfR, <span class="at">min.distance =</span> <span class="dv">500</span>)</span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>  <span class="er">|</span>                                                                      <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==</span>                                                                    <span class="er">|</span>   <span class="dv">3</span>%</span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====</span>                                                                  <span class="er">|</span>   <span class="dv">6</span>%</span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======</span>                                                                <span class="er">|</span>   <span class="dv">8</span>%</span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========</span>                                                              <span class="er">|</span>  <span class="dv">11</span>%</span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========</span>                                                            <span class="er">|</span>  <span class="dv">14</span>%</span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============</span>                                                          <span class="er">|</span>  <span class="dv">17</span>%</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============</span>                                                        <span class="er">|</span>  <span class="dv">19</span>%</span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================</span>                                                      <span class="er">|</span>  <span class="dv">22</span>%</span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-28"><a href="#cb37-28" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================</span>                                                    <span class="er">|</span>  <span class="dv">25</span>%</span>
<span id="cb37-29"><a href="#cb37-29" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-30"><a href="#cb37-30" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================</span>                                                   <span class="er">|</span>  <span class="dv">28</span>%</span>
<span id="cb37-31"><a href="#cb37-31" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-32"><a href="#cb37-32" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================</span>                                                 <span class="er">|</span>  <span class="dv">31</span>%</span>
<span id="cb37-33"><a href="#cb37-33" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-34"><a href="#cb37-34" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================</span>                                               <span class="er">|</span>  <span class="dv">33</span>%</span>
<span id="cb37-35"><a href="#cb37-35" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-36"><a href="#cb37-36" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================</span>                                             <span class="er">|</span>  <span class="dv">36</span>%</span>
<span id="cb37-37"><a href="#cb37-37" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-38"><a href="#cb37-38" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================</span>                                           <span class="er">|</span>  <span class="dv">39</span>%</span>
<span id="cb37-39"><a href="#cb37-39" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-40"><a href="#cb37-40" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================</span>                                         <span class="er">|</span>  <span class="dv">42</span>%</span>
<span id="cb37-41"><a href="#cb37-41" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-42"><a href="#cb37-42" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================</span>                                       <span class="er">|</span>  <span class="dv">44</span>%</span>
<span id="cb37-43"><a href="#cb37-43" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-44"><a href="#cb37-44" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================</span>                                     <span class="er">|</span>  <span class="dv">47</span>%</span>
<span id="cb37-45"><a href="#cb37-45" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-46"><a href="#cb37-46" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================</span>                                   <span class="er">|</span>  <span class="dv">50</span>%</span>
<span id="cb37-47"><a href="#cb37-47" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-48"><a href="#cb37-48" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=====================================</span>                                 <span class="er">|</span>  <span class="dv">53</span>%</span>
<span id="cb37-49"><a href="#cb37-49" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-50"><a href="#cb37-50" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=======================================</span>                               <span class="er">|</span>  <span class="dv">56</span>%</span>
<span id="cb37-51"><a href="#cb37-51" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-52"><a href="#cb37-52" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=========================================</span>                             <span class="er">|</span>  <span class="dv">58</span>%</span>
<span id="cb37-53"><a href="#cb37-53" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-54"><a href="#cb37-54" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===========================================</span>                           <span class="er">|</span>  <span class="dv">61</span>%</span>
<span id="cb37-55"><a href="#cb37-55" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-56"><a href="#cb37-56" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=============================================</span>                         <span class="er">|</span>  <span class="dv">64</span>%</span>
<span id="cb37-57"><a href="#cb37-57" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-58"><a href="#cb37-58" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===============================================</span>                       <span class="er">|</span>  <span class="dv">67</span>%</span>
<span id="cb37-59"><a href="#cb37-59" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-60"><a href="#cb37-60" aria-hidden="true" tabindex="-1"></a>  <span class="er">|=================================================</span>                     <span class="er">|</span>  <span class="dv">69</span>%</span>
<span id="cb37-61"><a href="#cb37-61" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-62"><a href="#cb37-62" aria-hidden="true" tabindex="-1"></a>  <span class="er">|===================================================</span>                   <span class="er">|</span>  <span class="dv">72</span>%</span>
<span id="cb37-63"><a href="#cb37-63" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-64"><a href="#cb37-64" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================</span>                  <span class="er">|</span>  <span class="dv">75</span>%</span>
<span id="cb37-65"><a href="#cb37-65" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-66"><a href="#cb37-66" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================</span>                <span class="er">|</span>  <span class="dv">78</span>%</span>
<span id="cb37-67"><a href="#cb37-67" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-68"><a href="#cb37-68" aria-hidden="true" tabindex="-1"></a>  <span class="er">|========================================================</span>              <span class="er">|</span>  <span class="dv">81</span>%</span>
<span id="cb37-69"><a href="#cb37-69" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-70"><a href="#cb37-70" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==========================================================</span>            <span class="er">|</span>  <span class="dv">83</span>%</span>
<span id="cb37-71"><a href="#cb37-71" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-72"><a href="#cb37-72" aria-hidden="true" tabindex="-1"></a>  <span class="er">|============================================================</span>          <span class="er">|</span>  <span class="dv">86</span>%</span>
<span id="cb37-73"><a href="#cb37-73" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-74"><a href="#cb37-74" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==============================================================</span>        <span class="er">|</span>  <span class="dv">89</span>%</span>
<span id="cb37-75"><a href="#cb37-75" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-76"><a href="#cb37-76" aria-hidden="true" tabindex="-1"></a>  <span class="er">|================================================================</span>      <span class="er">|</span>  <span class="dv">92</span>%</span>
<span id="cb37-77"><a href="#cb37-77" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-78"><a href="#cb37-78" aria-hidden="true" tabindex="-1"></a>  <span class="er">|==================================================================</span>    <span class="er">|</span>  <span class="dv">94</span>%</span>
<span id="cb37-79"><a href="#cb37-79" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-80"><a href="#cb37-80" aria-hidden="true" tabindex="-1"></a>  <span class="er">|====================================================================</span>  <span class="er">|</span>  <span class="dv">97</span>%</span>
<span id="cb37-81"><a href="#cb37-81" aria-hidden="true" tabindex="-1"></a>  <span class="sc">|</span>                                                                            </span>
<span id="cb37-82"><a href="#cb37-82" aria-hidden="true" tabindex="-1"></a>  <span class="er">|======================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb37-83"><a href="#cb37-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2803 out of 16307 input SNPs were not located within 500 base-pairs of another SNP and were retained despite filtering</span></span>
<span id="cb37-84"><a href="#cb37-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-85"><a href="#cb37-85" aria-hidden="true" tabindex="-1"></a><span class="co">#write out thinned vcf</span></span>
<span id="cb37-86"><a href="#cb37-86" aria-hidden="true" tabindex="-1"></a>vcfR<span class="sc">::</span><span class="fu">write.vcf</span>(vcfR.thin, <span class="st">"~/Downloads/aphelocoma.filtered.thinned.vcf.gz"</span>)</span></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
