<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Vizualise missing data per SNP, remove SNPs above a missing data cutoff — missing_by_snp • SNPfiltR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Vizualise missing data per SNP, remove SNPs above a missing data cutoff — missing_by_snp" />
<meta property="og:description" content="This function can be run in two ways: 1) Without 'cutoff' specified. This will vizualise the
amount of missing data in each sample across a variety of potential missing data cutoffs.
Additionally, it will show you dotplots visualizing the number of total SNPs retained across
a variety of filtering cutoffs, and the total proportion of missing data.
Based on these visualizations, you can make an informed decision on what you think might be an
optimal cutoff to minimize the overall missingness of your dataset while still retaining an
appropriate amount of SNPs for the downstream inferences you hope to make 2) with 'cutoff'
specified. This option will show you the dotplots with the cutoff you set,
and then remove SNPs above the missing data cutoff." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-example-vignette.html">scrub-jay-example-vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Vizualise missing data per SNP, remove SNPs above a missing data cutoff</h1>
    
    <div class="hidden name"><code>missing_by_snp.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function can be run in two ways: 1) Without 'cutoff' specified. This will vizualise the
amount of missing data in each sample across a variety of potential missing data cutoffs.
Additionally, it will show you dotplots visualizing the number of total SNPs retained across
a variety of filtering cutoffs, and the total proportion of missing data.
Based on these visualizations, you can make an informed decision on what you think might be an
optimal cutoff to minimize the overall missingness of your dataset while still retaining an
appropriate amount of SNPs for the downstream inferences you hope to make 2) with 'cutoff'
specified. This option will show you the dotplots with the cutoff you set,
and then remove SNPs above the missing data cutoff.</p>
    </div>

    <pre class="usage"><span class='fu'>missing_by_snp</span><span class='op'>(</span><span class='va'>vcfR</span>, cutoff <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>vcfR</th>
      <td><p>a vcfR object</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>a numeric value between 0-1 specifying the maximum proportion of missing data
allowed in a SNP to be retained for downstream analyses</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>if 'cutoff' is not specified, will return a dataframe containing the proportion
missing data and the total SNPs retained across each filtering level. If 'cutoff' is
specified, SNPs falling above the missing data cutoff will be removed,
and the filtered vcfR object will be returned.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>missing_by_snp</span><span class='op'>(</span>vcfR <span class='op'>=</span> <span class='fu'>SNPfiltR</span><span class='fu'>::</span><span class='va'><a href='vcfR.example.html'>vcfR.example</a></span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "cutoff is not specified, exploratory visualizations will be generated"</div><div class='output co'>#&gt; <span class='message'>Picking joint bandwidth of 0.0903</span></div><div class='img'><img src='missing_by_snp-1.png' alt='' width='700' height='433' /></div><div class='img'><img src='missing_by_snp-2.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt;    filt missingness snps.retained
#&gt; 1  0.30  0.28741007           417
#&gt; 2  0.50  0.21399417           343
#&gt; 3  0.60  0.17118644           295
#&gt; 4  0.65  0.14907063           269
#&gt; 5  0.70  0.12938776           245
#&gt; 6  0.75  0.11585903           227
#&gt; 7  0.80  0.09698492           199
#&gt; 8  0.85  0.08011696           171
#&gt; 9  0.90  0.06082090           134
#&gt; 10 0.95  0.04414894            94
#&gt; 11 1.00  0.00000000            11</div><div class='input'><span class='fu'>missing_by_snp</span><span class='op'>(</span>vcfR <span class='op'>=</span> <span class='fu'>SNPfiltR</span><span class='fu'>::</span><span class='va'><a href='vcfR.example.html'>vcfR.example</a></span>, cutoff <span class='op'>=</span> <span class='fl'>.6</span><span class='op'>)</span>
</div><div class='output co'>#&gt; [1] "cutoff is specified, filtered vcfR object will be returned"</div><div class='img'><img src='missing_by_snp-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; [1] "41% of SNPs fell below a completeness cutoff of 0.6 and were removed from the VCF"</div><div class='output co'>#&gt; ***** Object of Class vcfR *****
#&gt; 20 samples
#&gt; 1 CHROMs
#&gt; 295 variants
#&gt; Object size: 0.5 Mb
#&gt; 17.12 percent missing data
#&gt; *****        *****         *****</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


