<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Vizualise missing data per sample, remove samples above a missing data cutoff — missing_by_sample • SNPfiltR</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Vizualise missing data per sample, remove samples above a missing data cutoff — missing_by_sample" />
<meta property="og:description" content="This function can be run in two ways: 1) Without 'cutoff' specified. This will vizualise the
amount of missing data in each sample across a variety of potential missing data cutoffs.
Additionally, it will show you a dotplot ordering the amount of overall missing data
in each sample. Based on these visualizations, you can make an informed decision on what
you think might be an optimal cutoff to remove samples that are missing too much data to
be retained for downstream analyses. 2) with 'cutoff' specified.
This option will show you the dotplot with the cutoff you set, and then
remove samples above the missing data cutoff you set, and return the filtered vcf to you." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SNPfiltR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/accuracy-benchmarking.html">Accuracy benchmarking</a>
    </li>
    <li>
      <a href="../articles/performance-benchmarking.html">Performance benchmarking</a>
    </li>
    <li>
      <a href="../articles/reproducible-vignette.html">reproducible-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-RADseq-vignette.html">scrub-jay-RADseq-vignette</a>
    </li>
    <li>
      <a href="../articles/scrub-jay-UCE-vignette.html">scrub-jay-UCE-vignette</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Vizualise missing data per sample, remove samples above a missing data cutoff</h1>
    
    <div class="hidden name"><code>missing_by_sample.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function can be run in two ways: 1) Without 'cutoff' specified. This will vizualise the
amount of missing data in each sample across a variety of potential missing data cutoffs.
Additionally, it will show you a dotplot ordering the amount of overall missing data
in each sample. Based on these visualizations, you can make an informed decision on what
you think might be an optimal cutoff to remove samples that are missing too much data to
be retained for downstream analyses. 2) with 'cutoff' specified.
This option will show you the dotplot with the cutoff you set, and then
remove samples above the missing data cutoff you set, and return the filtered vcf to you.</p>
    </div>

    <pre class="usage"><span class='fu'>missing_by_sample</span><span class='op'>(</span><span class='va'>vcfR</span>, popmap <span class='op'>=</span> <span class='cn'>NULL</span>, cutoff <span class='op'>=</span> <span class='cn'>NULL</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>vcfR</th>
      <td><p>a vcfR object</p></td>
    </tr>
    <tr>
      <th>popmap</th>
      <td><p>if specifies, it must be a two column dataframe with columns names 'id' and 'pop'.
IDs must match the IDs in the vcfR object</p></td>
    </tr>
    <tr>
      <th>cutoff</th>
      <td><p>a numeric value between 0-1 specifying the maximum proportion of missing data
allowed in a sample to be retained for downstream analyses</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>if 'cutoff' is not specified, will return a dataframe containing the average
depth and proportion missing data in each sample. If 'cutoff' is specified, the samples
falling above the missing data cutoff
will be removed, and the filtered vcfR object will be returned.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Note: This decision is highly project specific, but these visualizations should help
you get a feel for how very low data samples cannot be rescued simply by a missing
data SNP filter. If you want to remove specific samples from your vcf that cannot
be specified with a simple cutoff refer to this great
<a href='https://knausb.github.io/vcfR_documentation/sequence_coverage.html'>tutorial</a>
which is the basis for the code underlying this function.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>missing_by_sample</span><span class='op'>(</span>vcfR <span class='op'>=</span> <span class='fu'>SNPfiltR</span><span class='fu'>::</span><span class='va'><a href='vcfR.example.html'>vcfR.example</a></span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>No popmap provided</span></div><div class='img'><img src='missing_by_sample-1.png' alt='' width='700' height='433' /></div><div class='img'><img src='missing_by_sample-2.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt;                     indiv filt snps.retained
#&gt; 1    A_californica_333849  0.5           334
#&gt; 2    A_californica_333854  0.5           307
#&gt; 3    A_californica_333855  0.5           295
#&gt; 4    A_californica_333857  0.5           324
#&gt; 5    A_californica_333860  0.5           300
#&gt; 6      A_insularis_334031  0.5           313
#&gt; 7      A_insularis_334032  0.5           330
#&gt; 8      A_insularis_334033  0.5           327
#&gt; 9      A_insularis_334034  0.5           314
#&gt; 10     A_insularis_334037  0.5           317
#&gt; 11   A_woodhouseii_334133  0.5           312
#&gt; 12   A_woodhouseii_334134  0.5           208
#&gt; 13   A_woodhouseii_334142  0.5           300
#&gt; 14   A_woodhouseii_334148  0.5           190
#&gt; 15   A_woodhouseii_334153  0.5           285
#&gt; 16  A_coerulescens_396256  0.5           247
#&gt; 17  A_coerulescens_396259  0.5           143
#&gt; 18  A_coerulescens_396262  0.5            18
#&gt; 19  A_coerulescens_396263  0.5           215
#&gt; 20  A_coerulescens_396264  0.5           313
#&gt; 21   A_californica_333849  0.6           291
#&gt; 22   A_californica_333854  0.6           275
#&gt; 23   A_californica_333855  0.6           268
#&gt; 24   A_californica_333857  0.6           284
#&gt; 25   A_californica_333860  0.6           275
#&gt; 26     A_insularis_334031  0.6           281
#&gt; 27     A_insularis_334032  0.6           291
#&gt; 28     A_insularis_334033  0.6           285
#&gt; 29     A_insularis_334034  0.6           278
#&gt; 30     A_insularis_334037  0.6           282
#&gt; 31   A_woodhouseii_334133  0.6           281
#&gt; 32   A_woodhouseii_334134  0.6           199
#&gt; 33   A_woodhouseii_334142  0.6           275
#&gt; 34   A_woodhouseii_334148  0.6           182
#&gt; 35   A_woodhouseii_334153  0.6           259
#&gt; 36  A_coerulescens_396256  0.6           237
#&gt; 37  A_coerulescens_396259  0.6           138
#&gt; 38  A_coerulescens_396262  0.6            18
#&gt; 39  A_coerulescens_396263  0.6           211
#&gt; 40  A_coerulescens_396264  0.6           280
#&gt; 41   A_californica_333849  0.7           244
#&gt; 42   A_californica_333854  0.7           234
#&gt; 43   A_californica_333855  0.7           238
#&gt; 44   A_californica_333857  0.7           238
#&gt; 45   A_californica_333860  0.7           236
#&gt; 46     A_insularis_334031  0.7           238
#&gt; 47     A_insularis_334032  0.7           245
#&gt; 48     A_insularis_334033  0.7           238
#&gt; 49     A_insularis_334034  0.7           240
#&gt; 50     A_insularis_334037  0.7           237
#&gt; 51   A_woodhouseii_334133  0.7           239
#&gt; 52   A_woodhouseii_334134  0.7           192
#&gt; 53   A_woodhouseii_334142  0.7           241
#&gt; 54   A_woodhouseii_334148  0.7           171
#&gt; 55   A_woodhouseii_334153  0.7           230
#&gt; 56  A_coerulescens_396256  0.7           218
#&gt; 57  A_coerulescens_396259  0.7           135
#&gt; 58  A_coerulescens_396262  0.7            18
#&gt; 59  A_coerulescens_396263  0.7           199
#&gt; 60  A_coerulescens_396264  0.7           235
#&gt; 61   A_californica_333849  0.8           170
#&gt; 62   A_californica_333854  0.8           170
#&gt; 63   A_californica_333855  0.8           168
#&gt; 64   A_californica_333857  0.8           171
#&gt; 65   A_californica_333860  0.8           170
#&gt; 66     A_insularis_334031  0.8           170
#&gt; 67     A_insularis_334032  0.8           171
#&gt; 68     A_insularis_334033  0.8           170
#&gt; 69     A_insularis_334034  0.8           170
#&gt; 70     A_insularis_334037  0.8           171
#&gt; 71   A_woodhouseii_334133  0.8           171
#&gt; 72   A_woodhouseii_334134  0.8           158
#&gt; 73   A_woodhouseii_334142  0.8           171
#&gt; 74   A_woodhouseii_334148  0.8           151
#&gt; 75   A_woodhouseii_334153  0.8           169
#&gt; 76  A_coerulescens_396256  0.8           166
#&gt; 77  A_coerulescens_396259  0.8           115
#&gt; 78  A_coerulescens_396262  0.8            18
#&gt; 79  A_coerulescens_396263  0.8           156
#&gt; 80  A_coerulescens_396264  0.8           170
#&gt; 81   A_californica_333849  0.9            94
#&gt; 82   A_californica_333854  0.9            94
#&gt; 83   A_californica_333855  0.9            94
#&gt; 84   A_californica_333857  0.9            94
#&gt; 85   A_californica_333860  0.9            94
#&gt; 86     A_insularis_334031  0.9            94
#&gt; 87     A_insularis_334032  0.9            94
#&gt; 88     A_insularis_334033  0.9            94
#&gt; 89     A_insularis_334034  0.9            94
#&gt; 90     A_insularis_334037  0.9            94
#&gt; 91   A_woodhouseii_334133  0.9            94
#&gt; 92   A_woodhouseii_334134  0.9            93
#&gt; 93   A_woodhouseii_334142  0.9            94
#&gt; 94   A_woodhouseii_334148  0.9            94
#&gt; 95   A_woodhouseii_334153  0.9            94
#&gt; 96  A_coerulescens_396256  0.9            94
#&gt; 97  A_coerulescens_396259  0.9            90
#&gt; 98  A_coerulescens_396262  0.9            16
#&gt; 99  A_coerulescens_396263  0.9            94
#&gt; 100 A_coerulescens_396264  0.9            94
#&gt; 101  A_californica_333849  1.0            11
#&gt; 102  A_californica_333854  1.0            11
#&gt; 103  A_californica_333855  1.0            11
#&gt; 104  A_californica_333857  1.0            11
#&gt; 105  A_californica_333860  1.0            11
#&gt; 106    A_insularis_334031  1.0            11
#&gt; 107    A_insularis_334032  1.0            11
#&gt; 108    A_insularis_334033  1.0            11
#&gt; 109    A_insularis_334034  1.0            11
#&gt; 110    A_insularis_334037  1.0            11
#&gt; 111  A_woodhouseii_334133  1.0            11
#&gt; 112  A_woodhouseii_334134  1.0            11
#&gt; 113  A_woodhouseii_334142  1.0            11
#&gt; 114  A_woodhouseii_334148  1.0            11
#&gt; 115  A_woodhouseii_334153  1.0            11
#&gt; 116 A_coerulescens_396256  1.0            11
#&gt; 117 A_coerulescens_396259  1.0            11
#&gt; 118 A_coerulescens_396262  1.0            11
#&gt; 119 A_coerulescens_396263  1.0            11
#&gt; 120 A_coerulescens_396264  1.0            11</div><div class='input'><span class='fu'>missing_by_sample</span><span class='op'>(</span>vcfR <span class='op'>=</span> <span class='fu'>SNPfiltR</span><span class='fu'>::</span><span class='va'><a href='vcfR.example.html'>vcfR.example</a></span>, cutoff <span class='op'>=</span> <span class='fl'>.7</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>1 samples are above a 0.7 missing data cutoff, and were removed from VCF</span></div><div class='img'><img src='missing_by_sample-3.png' alt='' width='700' height='433' /></div><div class='output co'>#&gt; ***** Object of Class vcfR *****
#&gt; 19 samples
#&gt; 1 CHROMs
#&gt; 500 variants
#&gt; Object size: 0.7 Mb
#&gt; 35 percent missing data
#&gt; *****        *****         *****</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Devon DeRaad.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


